\documentclass[12pt]{pnas-new}
%\documentclass[draft]{pnas-new}
% Use the lineno option to display guide line numbers if required.
% Note that the use of elements such as single-column equations
% may affect the guide line number alignment. 

\newcommand{\beginsupplement}{%
        \setcounter{table}{0}
        \renewcommand{\thetable}{S\arabic{table}}%
        \setcounter{figure}{0}
        \renewcommand{\thefigure}{S\arabic{figure}}%
     }
\newcommand{\sgcomment}[1]{{\textcolor{blue}{SG: #1}}}
\newcommand{\adpcomment}[1]{{\textcolor{orange}{ADP: #1}}}
%\usepackage{subfig}
\usepackage{adjustbox}
\usepackage{subcaption}
\usepackage{graphicx}
\usepackage{float}
\usepackage{dblfloatfix}
\usepackage[switch]{lineno}
\usepackage{comment}
%\usepackage[section]{placeins}

\linenumbers

\templatetype{pnasresearcharticle} % Choose template
\title{UMAP reveals cryptic population structure and phenotype heterogeneity in large genomic cohorts}

% Use letters for affiliations, numbers to show equal authorship (if applicable) and to indicate the corresponding author
\author[a,b]{Alex Diaz-Papkovich}
\author[b,c]{Luke Anderson-Trocm\'e}
\author[b,c]{Chief Ben-Eghan}
\author[b,c,1]{Simon Gravel} 
\affil[a]{Department of Quantitative Life Sciences, McGill University, Montreal, QC, H3A 0G1 Canada}
\affil[b]{McGill University and Genome Quebec Innovation Centre, Montreal, QC, H3A 0G1, Canada}
\affil[c]{Department of Human Genetics, McGill University, Montreal, QC, H3A 0G1, Canada. \textsuperscript{1}To whom correspondence should be addressed. E-mail: simon.gravel@mcgill.ca}

% Please include corresponding author, author contribution and author declaration information
\correspondingauthor{}

% Please give the surname of the lead author for the running footer
\leadauthor{Diaz-Papkovich} 

%An Analysis is a new analysis of existing data (typically large genomic, transcriptomic or proteomic data sets from arrays or other high-throughput platforms) or new data obtained in a comparative analysis of technologies that lead to novel and arresting conclusions of importance to a broad audience. The main text (excluding abstract, Methods, references and figure legends) is approximately 3,000 words. The abstract is typically 100-150 words, unreferenced. Analyses have no more than 6 display items (figures and/or tables). An introduction (without heading) is followed by sections headed Results, Discussion and online Methods. The Results and online Methods should be divided by topical subheadings; the Discussion does not contain subheadings. As a guideline, Analysis allow up to 50 references.

\begin{abstract}
{\large Abstract.}
%The abstract is typically 100-150 words, unreferenced. 
Improvements in sequencing have generated massive and diverse genomic datasets of populations. Dimension reduction is necessary to condense and analyze the data. Uniform manifold approximation and projection (UMAP) is a non-linear dimension reduction tool recently made available. Here we apply UMAP to three well-studied genotype datasets and discover overlooked subpopulations within the American Hispanic population, fine-scale relationships between geography, genotypes, and phenotypes in the UK population, and cryptic structure in the Thousand Genomes Project data. We demonstrate this approach outperforms other standard methods, is well-suited to the influx of large and diverse data, and opens new lines of inquiry in population-scale datasets.
\end{abstract}

\begin{document}

% Optional adjustment to line up main text (after abstract) of first page with line numbers, when using both lineno and twocolumn options.
\verticaladjustment{-2pt}

\maketitle
\thispagestyle{firststyle}
\ifthenelse{\boolean{shortarticle}}{\ifthenelse{\boolean{singlecolumn}}{\abscontentformatted}{\abscontent}}{}
Questions in medicine, anthropology, and related fields hinge on interpreting the deluge of genomic data provided by modern high-throughput sequencing technologies. Because genomic datasets are high-dimensional, their interpretation requires statistical methods that can comprehensively condense information in a manner that is understandable to researchers and minimizes the amount of data that is sacrificed. Both model-based and model-agnostic approaches to summarize data have played important roles in shaping our understanding of the evolution of our species~\cite{lawson2012inference}.

Here we will focus on nonparametric approaches to visualize relatedness patterns among individuals within populations. If we consider unphased single nucleotide polymorphism (SNP) data, an individual genome can be represented as a sequence of integers corresponding to the number of alleles carried by the individual at each of the $L$ SNPs for which genotypes are available, with $L$ typically larger than $100,000$. Since each individual is represented as an $L$-dimensional vector, dimension reduction methods are needed to visualize the data.

Principal component analysis (PCA) is often the first dimensional reduction tool used for genomic data. It identifies and ranks directions in genotype space that explain most-to-least variance among individuals. Positions of individuals along directions of highest variance can then be used to summarize individual genotypes. PCA coordinates have natural genealogical interpretations in terms of expected times to a most recent common ancestor (TMRCA) \cite{mcvean2009genealogical}, and are used empirically to reveal admixture \cite{brisbin2012pcadmix}, continuous isolation-by-distance \cite{novembre2008europe, nelson2008population}, as well as technical artefacts. PCA coordinates are particularly well-suited to correct for population structure in GWAS~\cite{eigen2006}.

\begin{figure*}[!ht]
    \centering
    \includegraphics[width=0.7\textwidth]{images/montage_1000G_labels_axes.pdf}
    \caption{Four methods of dimension reduction of 1KGP genotype data with population labels (i) PCA maps individuals in a triangle with vertices corresponding to African, Asian/Native American, and European continental ancestry. Discarding lower-variance PCs leads to overlap of populations with no close affinity, such as Central and South American populations with South Asians. (ii) t-SNE forms groups corresponding to continents, with some overlap between European and Central and South American people. Smaller subgroups are visible within continental clusters. The cloud of peripheral points results from the method's poor convergence. (iii) UMAP forms distinct clusters related to continent with clearly defined subgroups. Japanese, Finnish, Luhya, and some Punjabi and Telugu populations form separate clusters consistent with their population history\cite{10002015global}. (iv) UMAP on the first 15 principal components forms fine-scale clusters for individual populations. Groups closely related by ancestry or geography, such as African Caribbean/African American, Spanish/Italian, and Kinh/Dai populations cluster together. Results using t-SNE on principal components are presented in figure~\ref{fig:supp_megamontage_pc2_9}. Axes in UMAP and t-SNE are arbitrary. Since the algorithms prioritize local distances, long distances between clusters are not meaningful.
    ACB, African Caribbean in Barbados;
    ASW, African Ancestry in Southwest US;
    BEB, Bengali;
    CDX, Chinese Dai;
    CEU, Utah residents with Northern/Western European ancestry;
    CHB, Han Chinese;
    CHS, Southern Han Chinese;
    CLM, Colombian in Medellin, Colombia;
    ESN, Esan in Nigeria;
    FIN, Finnish;
    GBR, British in England and Scotland;
    GWD, Gambian;
    GTH, Gujarati;
    IBS, Iberian in Spain;
    ITU, Indian Telugu in the UK;
    JPT, Japanese;
    KHV, Kinh in Vietnam;
    LWK, Luhya in Kenya;
    MSL, Mende in Sierra Leone;
    MXL, Mexican in Los Angeles, California;
    PEL, Peruvian;
    PJL, Punjabi in Lahore, Pakistan;
    PUR, Puerto Rican;
    STU, Sri Lankan Tamil in the UK;
    TSI, Tuscani in Italy;
    YRI, Yoruba in Nigeria}
    \label{fig:1000g_compare}
\end{figure*}

The amount of information encoded in the lower-variance principal components increases with sample size, so researchers typically examine multiple two-dimensional projections to explore data. Some features may be hidden by the projections or hard to interpret. To display more of the high-dimensional features of the data in a two dimensional figure, we can use non-linear transformations that seek to preserve the local structure of the data. A popular method for visualization is t-distributed stochastic neighbour embedding (t-SNE)\cite{maaten2008visualizing}. t-SNE has been used before to visualize SNPs\cite{platzer2013visualization}; using data from the 1000 Genomes Project (1KGP)\cite{10002015global}, it groups individuals corresponding roughly to their continent of origin, with smaller ethnic sub-groups visible within the larger continental clusters\cite{li2017tsne}. However, t-SNE struggles with very large datasets, when a large number of locally optimal configurations make convergence to a globally satisfying solution difficult. 

Uniform Manifold Approximation and Projection (UMAP) is a dimension reduction technique designed to model and preserve the high-dimensional topology of data points in the low-dimensional space\cite{2018arXivUMAP}. With genotype data, UMAP creates a neighbourhood around each individual's genetic coordinates and identifies a pre-selected number of neighbours to build high-dimensional manifolds. The end result is a patchwork of low-dimensional representations of neighbourhoods that groups genetically similar individuals together on a local scale while better preserving long-range topological connections to more distantly related individuals. The method has been successfully applied to single-cell RNA sequencing datasets\cite{umap2018singlecell}.

A common practice in dimensional reduction is to first apply PCA. In addition to being computationally advantageous, this discards noise that can confound nonlinear approaches: population structure arising from $n$ isolated randomly-mating demes can be described by the leading $n-1$ PCs, with the following PCs describing stochastic variation in relatedness~\cite{eigen2006}. Selecting the leading PCs therefore has potential to extract meaningful population structure while filtering out stochastic noise. We explore different strategies to pre-process the data and investigate discrete and continuous population structure patterns present in large datasets of human genotypes: the 1KGP, the Health and Retirement Study (HRS)\cite{juster1995overview}, and the UK BioBank (UKBB)\cite{sudlow2015uk}, and compare UMAP's performance to t-SNE.  

\begin{figure*}
    \centering
    \begin{subfigure}{0.5\columnwidth}
    \includegraphics[width=\columnwidth]{images/HRS_1000G_NP1_UMAP_PC7_NC2_NN15_MD05_pca_hrshisp_added1kgp_2018115153245_1kgp_hisp.pdf}
    \caption{Subset of Hispanic population of the HRS}
    \label{fig:supp_umap_hrs_hisp_1kgp_hisp}
    \end{subfigure}\hfill
        \begin{subfigure}{0.5\columnwidth}
    \includegraphics[width=\columnwidth]{images/Asthma_FullAsian_PC_QC_new_UMAP_PC8_NC2_NN15_MD05_20193117303txt_eth_for_ms_lw2.jpeg}
    \caption{Subset of the Asian population of the UKBB}
        \label{fig:umap_ukbb_indian}
    \end{subfigure}
    \caption{Applying UMAP to subets of data can reveal deep population structure. (a) UMAP on the top $7$ principal components of the self-identified Hispanic population of the HRS reveals a cluster (highlighted). Colouring the points by birthplace shows they were born almost entirely in the Mountain region of the United States (New Mexico, Arizona, Colorado, Utah, Nevada, Wyoming, Idaho, and Montana). When populations from the 1KGP are projected onto the UMAP embedding they do not map to the cluster. Six 1KGP populations are presented:
    CLM, Colombian in Medellin, Colombia;
    IBS, Iberian in Spain;
    MXL, Mexican in Los Angeles, California;
    PEL, Peruvian;
    PUR, Puerto Rican;
    TSI, Tuscani in Italy. Figure~\ref{fig:supp_umap_hrs_hisp_admix} presents the same projection of individuals from the HRS coloured by estimated admixture proportions. (b) UMAP on the top $8$ principal components of the self-identified Asian populations of the UKBB create clusters. Almost all individuals in the highlighted cluster are Indian individuals born in Kenya.}
    \label{fig:subpops}
\end{figure*}

\subsection*{Results}
\subsubsection*{Fine-scale visualization of the 1KGP dataset} The 1KGP contains genotype data of 3,450 individuals from 26 relatively distinct labeled populations\cite{10002015global}. Figure~\ref{fig:1000g_compare} shows visualizations using PCA, t-SNE, UMAP, and UMAP with PCA pre-processing. Using UMAP and t-SNE on the genotype data presents clusters that are roughly grouped by continent, with UMAP showing a clear hierarchy of population and continental clusters, whereas t-SNE fails to assign many individuals to population clusters. Using either on the top principal components leads to distinct population clusters and less defined continental structure. Adding more components results in progressively finer clusters until approximately 20 populations appear using 15 components; adding further components converges to results similar to using the entire genotype data (see figures~\ref{fig:supp_megamontage_pc2_9}, \ref{fig:supp_megamontage_pc10_50}, and \ref{fig:supp_montage_1kgp_converge}). Patterns are also evident when using UMAP on the last components (\ref{fig:supp_1kgp_3350}).

Focusing on UMAP with 15 principal components (figure~\ref{fig:1000g_compare}~(iv)), several population clusters reflect shared ancestries. British individuals from England and Scotland form a cluster mixed with those from Utah who claim Northern and Western European ancestry. Toscani and Iberian individuals form a group reflecting their Mediterranean heritage. African Americans in the Southwest US, African Caribbean individuals in Barbados, and some Puerto Ricans also form a cluster. Three East Asian groups appear: one is largely Han and Southern Han individuals, another is comprised of the Chinese Dai in southern China and the Kinh from Vietnam, and the third is the Japanese population. Looser geographical groupings include Colombians and Peruvians, and the Esan and Yoruba populations of Nigeria; both groupings appear as connected sub-clusters. The South Asian populations also form a loose grouping. 

Only a few individuals cluster differently than the majority of individual bearing the same population label: a few Mexican individuals cluster with Spanish and Italian individuals, and a few Puerto Ricans cluster with the African Americans and African Caribbeans, likely resulting from ancestry proportions that differ from the majority. One Gambian-identified individual is present in a cluster that is otherwise entirely Mende people from Sierra Leone. Two populations form multiple clusters: Gujarati Indians in Houston, Texas and Punjabi people in Lahore, Pakistan. This clustering effect is robust to the number of components considered (see figure~\ref{fig:supp_megamontage_pc10_50}). Differentiation in the Gujarati population from substructure has been previously noted\cite{reich2009india}, and analysis of the cluster by 23andMe suggests it is the result of endogamy\cite{23andme}.

\subsubsection*{The genetic continuum of admixed populations} The 1KGP sampled individuals from relatively distinct populations, so the data are more likely to form clusters. Most medical cohorts, however, comprise larger numbers of individuals sampled across extended geographical areas. The HRS contains genotype data of 12,454 Americans from a variety of backgrounds. Using UMAP on the first 10 principal components, we demonstrate projections that present a collection of sub-populations and a continuum of genetic variation.

The HRS forms several large groupings and clusters, reflecting both ethnicity (figure~\ref{fig:supp_umap_hrs_eth}) and admixture proportions (figure~\ref{fig:umap_hrs_admix}). Gradients in admixture proportion are visible within the predominantly Hispanic cluster, but not within the predominantly Black cluster, perhaps because the variance in ancestry proportions is greater among Hispanics. The "White Not Hispanic" (WNH) group forms several interconnected clusters, and these do not correspond to broad geographical areas (figure~\ref{fig:supp_hrs_born}). The clarity of the interconnected clusters varies by parameterization, but they consistently form a large, roughly connected group.

To investigate possible ancestries related to populations in the 1KGP we took two approaches. In the first we generated PC axes and a UMAP embedding using HRS and 1KGP data together (figure~\ref{fig:supp_hrs_1000g}). In the second we used the PC axes and UMAP embedding generated in figure~\ref{fig:umap_hrs_admix} and projected 1KGP data onto it (figure~\ref{fig:supp_hrs_1kgp_projected}). Both approaches reveal substructure within the Hispanic cluster, groupings of Finnish individuals within the WNH groups, as well as Italian and Spanish individuals grouping near the White Hispanic population. One group of WNH individuals regularly appears at the periphery of the main cluster and does not cluster with any 1KGP populations.

\subsubsection*{Regional patterns in the Hispanic subpopulation} Applying UMAP to self-identified Hispanic individuals in the HRS reveals clear groupings related to birth region. One cluster, highlighted in figure~\ref{fig:supp_umap_hrs_hisp_1kgp_hisp}, consists almost entirely of individuals born in the Mountain Region of the United States. This cluster is not apparent when looking at a grid of pairwise plots of the first 8 principal components, provided in figure~\ref{fig:supp_hrs_hisp_grid}, as the signal is distributed along PCs 3, 4, and 6. Even though continental admixture patterns do correlate with UMAP position (figure~\ref{fig:supp_umap_hrs_hisp_admix}), these do not explain the Mountain Region cluster. Individuals from 1KGP populations do not appear in the cluster when projected to the UMAP embedding. The cluster possibly comprises the Hispano/Nuevomexicano population of the Southwest US, who have been present in the Mountain Region area long before the more recent immigrants from Latin America, and whose ancestry is expected to reflect both distinct Native ancestry and population-specific drift relative to other Hispanic populations. Such a grouping has been previously identified in AncestryDNA data using sequential clustering on identity-by-descent connections\cite{han2017clustering}; a recent preprint discusses the Mountain Region Hispanics with a more detailed historical description\cite{Jordan333609}.  

\subsubsection*{Population structure in the UKBB reflects local and global genetic variation} 
The UKBB is a trove of data on 488,377 individuals containing phenotypic measures and self-identified ethnic backgrounds. UMAP on the top 10 principal components reveals continuous and discrete population structure (figure~\ref{fig:umap_ukbb_pc10}): the patchwork of local topologies identifies continuous structure within the British population as well as admixture gradients despite the very unbalanced population sizes. The result is a comprehensive portrait of genetic variation capturing population relationships not visible using other methods, succinctly illustrating the complex structure of large and multi-ethnic datasets.

\begin{figure*}%
\centering
\begin{subfigure}{.5\columnwidth}
\includegraphics[width=\columnwidth]{images/UKBB_PC0_PC1_eth_crop.jpeg}%
\caption{Principal components 1 and 2}%
\label{fig:pc_ukbb}%
\end{subfigure}\hfill%
\begin{subfigure}{.5\columnwidth}
\includegraphics[width=\columnwidth]{images/UKBB_UMAP_PC10_NN15_MD05_eth_labels.pdf}%
\caption{UMAP on first 10 principal components}%
\label{fig:umap_ukbb_pc10}%
\end{subfigure}%
\caption{The UKBB coloured by self-reported ethnic background. (a) The first two principal components, showing the usual triangle with vertices corresponding to African, Asian/Native American, and European ancestries, and intermediate values indicating admixture or lack of relationship to the vertex populations. (b) UMAP on the first 10 principal components. The cluster of White British and White Irish individuals is greatly expanded, with the Irish forming a distinct sub cluster mixed with the White British population. South Asian and East Asian individuals form their separate clusters, as do individuals of African or Caribbean backgrounds. Population clusters are connected by "trails" comprised of large proportions of individuals with mixed backgrounds.
BA, Black African; 
BC, Black Caribbean; 
BG, Bangladeshi; 
CHN, Chinese; 
IND, Indian;
PK, Pakistani;
WB, White British;
WI, White Irish;
WBC, White and Black Caribbean; 
WBA, White and Black African; 
WAA, White and Asian;
AAB, Any other Asian Background; 
ABB, Any other Black Background;
AWB, Any other White Background;
AMB, Any other Mixed Background;
OEG, Other ethnic group. 
}
\label{fig:fig_ukbb}
\end{figure*}

The largest body in figure~\ref{fig:umap_ukbb_pc10} consists of the White British and Irish populations. The Irish population concentrates in one part of this group, but many individuals are also scattered throughout the British-identified population. Individuals identifying as Black African and Black Caribbean partially overlap, but admixed individuals form distinct trails leading to Asian and European clusters. Chinese individuals form a cluster, within a broader East Asian super-population; Indian, Pakistani, and Bangladeshi populations form a closely bound group as well. The East Asian and South Asian super-populations each have large clusters of individuals who identify as having an "other Asian background" or belonging to an "other ethnic group". The patchwork of genetic neighbourhoods is connected by trails of admixed individuals, which converge at a nexus of individuals with a variety of ethnicities. Many claim mixed ancestry, and there are clusters of individuals who belong to an "other ethnic group". Using data on countries of birth, we identified many groups in figure~\ref{fig:supp_ukbb_cob}, and confirmed they appeared in intuitive areas with, e.g., Japanese and Filipino groups being projected near Chinese groups.

Figure~\ref{fig:umap_ukbb_geo_osgb} presents the UMAP projection in figure~\ref{fig:umap_ukbb_pc10} coloured in by geographical coordinates from the Ordnance Survey National Grid (OSGB1936), with distances defined as a north or east position relative to the Isles of Scilly. Geographic clusters form in the large White British grouping, reflecting the relationship between genetic and geographic distance, as has been observed in Europe-wide data\cite{novembre2008europe}. Most admixture lines have coordinates in the south east of the UK, corresponding to the city of London and reflecting its high migrant population. The detailed shape of extended clusters is not stable as we vary the number of PCs included. Figures~\ref{fig:supp_montage_ukbb_eth}, \ref{fig:supp_montage_ukbb_ns}, and~\ref{fig:supp_montage_ukbb_ew} show UMAP plots using the top $40$ PCs from the UKBB. The shape of the White British cluster is different, and we observe finer patterns of geographic variation, yet the observations made earlier are maintained.

\begin{figure}
    \centering
    \begin{subfigure}{\columnwidth}
    \includegraphics[width=0.4\columnwidth]{images/UKBB_UMAP_PC10_NN15_MD05_2018328174511_ns_permuted_10nn_sd50000_base50000_201883118582.pdf}
    \includegraphics[width=.4\columnwidth]{images/UKBB_UMAP_PC10_NN15_MD05_2018328174511_ew_permuted_10nn_sd50000_base50000_201883118582.pdf}
    \caption{Colouring the projection by geographic coordinates.}
    \label{fig:umap_ukbb_geo_osgb}
    \end{subfigure}
     \begin{subfigure}{\columnwidth}
         \includegraphics[width=0.4\columnwidth]{images/UKBB_UMAP_PC10_NN15_MD05_2018328174511_2018714161841_Height_res_pct1_f.pdf}
    \includegraphics[width=.4\columnwidth]{images/UKBB_UMAP_PC10_NN15_MD05_2018328174511_201871416519_leukocyte_count_pct5_f.pdf}
   \caption{Colouring the projection by phenotypic data}
    \label{fig:umap_height_female}
      \end{subfigure}
      \caption{Using different colourings with the projection in figure~\ref{fig:umap_ukbb_pc10}, relationships between population structure, geography, and phenotypes become clear. (Top) Each individual is coloured by their geographical coordinates of residence. Coordinates follow the UKBB's OSGB1936 geographic grid system and represent distance from the Isles of Scilly, which lie southwest of Great Britain. The left image colours individuals by their north-south ("northing") coordinates, and the right image colours them by their east-west ("easting") coordinates. Adding more components creates finer clusters.  (figures~\ref{fig:supp_montage_ukbb_ns} and~\ref{fig:supp_montage_ukbb_ew}). Northing values were truncated between $100$km and $700$km, and easting values were truncated between $200$km and $600$km. (Bottom) Females from the same projection coloured by age-adjusted difference from mean population height (left) and leukocyte counts (right). Individuals with missing data were excluded. To protect participant privacy, data in these images has been randomized as explained in the materials and methods section.}
\end{figure}

As an alternate visualization of geography and genetic diversity, we performed a 3D UMAP projection and converted the UMAP coordinates into RGB values, allowing us to plot individuals on a map of Great Britain, emphasizing both spatial gradients of genetic relatedness and increased diversity in urban centers (figure~\ref{fig:umap_ukbb_geo_map}). The patterns in rural areas observed are similar to those reported in~\cite{leslie2015fine} using the haplotype-based CHROMOPAINTER on British individuals whose grandparents lived nearby. Using data about country of birth, we also generalized this approach to colour a world map in figure~\ref{fig:umap_world_map}.

%\begin{figure}
%    \centering
%    \includegraphics[width=0.5\columnwidth]{images/UKBB_mapsInMaps_2D3_permute50.jpg}
%    \caption{A map of Great Britain coloured by a 3D UMAP projection. Each individual is assigned a 3D RGB vector based on 3D UMAP coordinates (inset): Individuals who are closer to each other in the projection will appear to be closer in colour. Patterns in genetic similarity are visible in Scotland, South England, the East and West Midlands, and major urban centres. A flattened 2D view of of the 3D projection used for colouring is presented in the top right. To protect participant privacy, data has been randomized as explained in the materials and methods.}
%    \label{fig:umap_ukbb_geo_map}
%\end{figure}

\begin{figure}
    \centering
    \begin{subfigure}{\columnwidth}
    \includegraphics[width=0.5\columnwidth]{images/UKBB_mapsInMaps_2D3_permute50.jpg}
    \caption{Each point is an individual placed based on where they live. Patterns in genetic similarity are visible in Scotland, South England, North and South Wales, the East and West Midlands, and major urban centres.}
    \label{fig:umap_ukbb_geo_map}
    \end{subfigure}
     \begin{subfigure}{\columnwidth}
         \includegraphics[width=0.9\columnwidth]{images/meanWorldMap.jpg}
   \caption{Using the country of birth of individuals in the UKBB, we colour countries by the closeness in 3D UMAP space of those born there. Broad patterns of similarity appear in East Asia, South Asia, North African and the Middle East, West Africa, and South America. Vestiges of colonialism and related migrations are visible with, e.g., European ancestry in South Africa and South Asian ancestry in Kenya and Tanzania. Because of the large number of White British individuals born abroad, to avoid skewing the colour scale they were not included unless they were born in the UK, Europe, Australia, Canada, or the United States. Zoomed maps of East Asia, the Caribbean, and Europe are available in figures~\ref{fig:supp_umap_ukbb_asia}, \ref{fig:supp_umap_ukbb_car}, and \ref{fig:supp_umap_ukbb_eur}, respectively.}
    \label{fig:umap_world_map}
      \end{subfigure}
      \caption{Maps coloured by 3D UMAP projections of the top 20 principal components of the UKBB. Each individual is assigned a 3D RGB vector based on 3D UMAP coordinates. Individuals who are closer to each other in the projection will be closer in colour in the maps. More details on colouring, as well as randomization of points to protect participant privacy, are available in the materials and methods.}
\end{figure}

Similarly to UMAP, t-SNE applied to the UKBB data both displays diversity within the "White British" population and identifies clusters among other groups. However, it has three drawbacks: it is much slower, requiring 2.26 hours for its first thousand iterations alone on 10 principal components against UMAP's 14 minutes; it fails to find a global optimum, which results in a scattering of individuals and groups that are not stable across independent runs; and it does not identify continuity between different continental groups resulting from admixture (figure~\ref{fig:supp_ukbb_tsne}).     

\subsubsection*{Patterns in phenotype related to population structure} 
Covariates such as height and leukocyte count (figure~\ref{fig:umap_height_female}) and autoimmune and asthma-related measures  (figures~\ref{fig:supp_ukbb_basophill_f} to \ref{fig:supp_ukbb_neutrophill_m}) correlate strongly with both discrete and continuous population structure. Several populations in figure~\ref{fig:umap_height_female}, including South Asian, East Asian, African, and others have noticeably lower-than-average heights. More subtle patterns are also visible: the area of the projection in figures~\ref{fig:umap_ukbb_pc10} with the cluster of White Irish people appears more blue than the main body of White British individuals; an unpaired two sample t-test of self-identified White Irish and White British individuals reveals statistically significant differences in age-adjusted mean height between the populations, with British males being taller on average by 0.846cm (p-value $2.10\times 10^{-23}$) and British females by 0.763cm (p-value $3.65\times 10^{-23}$) (see figures~\ref{fig:supp_box_height_f} and \ref{fig:supp_box_height_m} for boxplots). Height differences between Irish and British populations have been previously observed but the direction of the difference is not consistent\cite{robinson2015population,komlos1994stature}.

Forced expiratory volume in 1 second (FEV1) (figures~\ref{fig:supp_ukbb_fev_f}, \ref{fig:supp_ukbb_fev_m}) also shows strong correlations with certain populations --- South Asian, African, and Caribbean --- having considerably lower measurements on average (see figures~\ref{fig:supp_box_fev_f} and \ref{fig:supp_box_fev_m} for boxplots and p-values). Notably, there appears to be a juncture in the admixture population, highlighted in figure~\ref{fig:supp_comparison_fev_afr}, where the distribution of FEV1 changes. This roughly corresponds to the transition from Black African/Caribbean individuals to those who identified having mixed backgrounds. Boxplots and statistical testing suggest that relative to White British populations FEV1 values are significantly lower for Black African and Black Caribbean populations, but not for White and Black Caribbean and White and Black African populations. Unidentified populations highlighted in figure~\ref{fig:supp_comparison_fev_chi_eur} suggest that one ethnic group close to the Chinese may have higher than average FEV1 values compared to the relatively low values of the Chinese themselves; using available country of birth data, most of the unidentified ethnic group were born in Japan. These results merit further investigation and underscore the exploratory value of UMAP --- PCA alone compresses too many populations together to see finer details (figures~\ref{fig:supp_ukbb_pca_height_res_f}, \ref{fig:supp_ukbb_pca_fev_f}), while t-SNE's clustering breaks the desirable continuity between and within populations~(figures~\ref{fig:supp_ukbb_tsne_height_res_f}, \ref{fig:supp_ukbb_tsne_fev_f}).

\subsubsection*{Comparing t-SNE and UMAP}
To compare UMAP to t-SNE and assess the role of convergence, we ran UMAP, Python's sklearn implementation of t-SNE, and a hybrid approach where the early exaggeration phase of t-SNE (typically the first 250 iterations) has been replaced by a full UMAP run (600 epochs for the 1KGP data and 230 epochs for the HRS and UKBB data). The three approaches produce similar results in smaller datasets such as the 1KGP and HRS, with the main differences being in run-time and some qualitative factors such as clusters being more interconnected when using UMAP (figures~\ref{fig:supp_tsne_umap_compare_1kgp},~\ref{fig:supp_tsne_umap_compare_hrs}). When applied to the UKBB, UMAP provides a more compelling visualization of the data compared to both versions of t-SNE. The hybrid approach also produces a better value for the t-SNE metric than the default t-SNE approach (figure~\ref{fig:supp_tsne_umap_compare_ukbb_graph}), confirming that the default t-SNE optimization fails to identify the global t-SNE minimum. The standard t-SNE initialization generated a large number of arbitrary clusters, while initializing t-SNE with UMAP preserved much of the large scale structure created by the UMAP embedding but again began arbitrarily creating clusters and breaking the projection's continuous structure (figure~\ref{fig:supp_tsne_umap_compare_ukbb}). An animation of t-SNE iterations starting from the UMAP embedding can be seen in the supplementary files. Given these results, we recommend UMAP over t-SNE for large and diverse genomic datasets.

\subsection*{Discussion}
UMAP comprehensively illustrates information --- phenotypic, geographic, and ancestral --- contained within genotypes on fine-scale levels and within the context of a global population structure. Using it is straightforward and fast; given PCA data and a desktop computer, UMAP can be performed in 15 to 25 minutes on a sample of hundreds of thousands of individuals over tens of dimensions. It excels with larger datasets containing individuals with admixed backgrounds, which present discrete and continuous population structure. By patching together many local genetic neighbourhoods, the visualizations have continuous flow and a semblance of a global structure, unlike arbitrary clusters generated by t-SNE as in figures \ref{fig:supp_tsne_umap_compare_ukbb} and \ref{fig:supp_ukbb_tsne}. These are undesirable when we are interested in gradients across populations.

Using UMAP reveals intriguing clusters that would otherwise have been very difficult to identify via numerous pairwise PCA plots, such as the geographically restricted cluster within the Hispanic population of the HRS. The 1KGP --- frequently used in medical and population studies --- contained splits in the Gujarati and Punjabi population samples that were not visible through PCA or Admixture analysis alone (although a split among Gujarati is arguably visible in the Admixture analysis with K=12 in \cite{10002015global}). Application to the UKBB underscores the strength of UMAP in large cohorts, with fine relationships between genotype and phenotype and geography presented in visualizations that account for natural genetic clustering. Figures~\ref{fig:umap_height_female}, \ref{fig:supp_ukbb_fev_f}, and \ref{fig:supp_ukbb_zoom} demonstrate phenotypic variation within and across clusters, with height showing continuous variation across admixture edges, as expected from genetically controlled traits, and others, such as leukocyte counts or FEV1, showing sharper boundaries, as expected from environmentally determined traits.  

UMAP works well at highlighting groups of individuals who share more ancestry; when clusters form consistently, this suggests the individuals within it share more ancestry, though nothing can be said about the differences between clusters. The 1KGP dataset was deliberately crafted to come from geographically separated populations that are homogeneous internally. Thus, most of its populations form isolated clusters, with the exception of those that most closely share ancestries.

Like most non-linear methods, UMAP lacks direct interpretability. It is premised on local distances being more important than global distances; while points that appear very close locally can be thought to be more similar to each other, points that are distant from each other cannot be considered to be more different from each other. Figure~\ref{fig:supp_megamontage_pc10_50} demonstrates that disconnected population clusters can bounce around in each projection. For this reason, UMAP coordinates should not be used as GWAS covariates or for calculating distances between populations. The number of principal components used will have an effect on the visualization depending on the variable of interest. Across datasets, groupings on ethnicity formed slowly until reaching a stable number around 10 to 15 components. Geographical patterns in the UKBB continued to appear even up to 40 components, as visible in figures~\ref{fig:supp_montage_ukbb_ns} and \ref{fig:supp_montage_ukbb_ew}. The main limitation on how many principal components to use is computational power.

Both UMAP and t-SNE are sensitive to sample size, and spend more visual real estate for larger sample sizes than PCA. This is useful to identify significant patterns in a cohort, but it makes comparing visualization across cohorts difficult and exaggerates the genetic variation within a population, such as the White British population in the UKBB. We did not assign meaning to wiggles in UMAP figures, which occurred consistently in the UKBB but may be an artefact of the dimensional reduction strategy rather than a meaningful feature of the data. Hand-waving interpretations of pretty plots has a history of getting population geneticists in trouble (as pointed out, e.g., in \cite{novembre2008interpreting}): visualization is not a replacement for statistical testing.

With these caveats in mind, a priori data visualization plays a central role in quality control, hypothesis generation, and confounder identification for a wide range of genomic applications. Nonlinear approaches, despite their limitations, become increasingly useful as the size of datasets increases. UMAP, in particular, reveals a wide range of features that would not be apparent using linear maps. Given its ease of use, breadth of results, and low computational cost, we propose that UMAP should become a default companion to PCA in large genomic cohorts.

\matmethods{We used genotype data from 12,454 individuals from the Health and Retirement Study (HRS), genotyped on the Illumina Human Omni 2.5M platform\cite{juster1995overview}. Principal components were computed in PLINK v1.90b5.2 64-bit\cite{purcell2007plink} using variants with a minor allele frequency greater than 0.05, Hardy-Weinberg p-value of more than $1\times 10^{-6}$, and genotype missing rate of less than $0.1$, and sample with genotype missing rate of less than $0.1$. We used the principal components of genotype data from 488,377 individuals in the UK BioBank (UKBB) as computed by the cohort \cite{sudlow2015uk}. We used genotype data from 3,450 individuals from the 1KGP project using Affy 6.0 genotyping\cite{10002015global}.

The HRS contains genotype data of 12,454 American individuals across all 50 states who have provided racial identity (10,434 White, 1,652 Black, 368 Other) as well as whether they identify as Hispanic (1,203 total) and, if so, whether they identify as Mexican-American (705 total)\cite{juster1995overview}. We crossed these three variables to form a composite self-reported ethnicity resulting in 10 categories (e.g. White Hispanic Mexican-American), and considered birth regions based on the 10 census regions and divisions used by the US Census Bureau. Admixture proportions for each individual were estimated in \cite{baharian2016great} by assuming ancestral African, Asian/Native American, and European populations using RFMIX \cite{Maples:2013fia}. We have scaled each of the three proportions to values between 0 and 255 (with 100\% corresponding to 255), to colour individual points by their estimated admixture represented by RGB where red, green, and blue respectively correspond to African, European, and Asian/Native American ancestry.

The UKBB provides genotype data on 488,377 individuals along with self-identified ethnic background in a hierarchical tree-structured dictionary. Participants provided ethnic background on two occasions. We used the initial ethnicity after finding minimal differences between the two. The dataset is majority White (88.3\% British, 2.6\% Irish, 3.4\% other), with large populations identifying as Black (1.6\% either African, Caribbean, or other), Asian (1.9\% either Indian, Pakistani, Bangladeshi, or other), Chinese (0.3\%), an other ethnic group (0.8\%), mixed ethnicity (0.6\%), or an unavailable response (0.5\%). 

Scripts for all tests and plotting functions can be found on \url{https://github.com/diazale/gt-dimred}. A demo version using freely available 1KGP data is available at \url{https://github.com/diazale/1KGP_dimred}. PCA and standard t-SNE were done with Scikit-learn\cite{scikit-learn}. UMAP was performed using a Python implementation\cite{2018arXivUMAP}. Statistical testing was done in SciPy\cite{scipy} and StatsModels\cite{seabold2010statsmodels}.  

Both UMAP and t-SNE feature a number of adjustable parameters. Among the parameters that we varied, the number of PCs used in pre-processing of the data has the largest effect for both methods (see figures~\ref{fig:supp_megamontage_pc2_9} and \ref{fig:supp_megamontage_pc10_50}). 

We tested different choices for perplexity in t-SNE. The default value of 30 provided comparable performance to other parameter choices.  Similarly, we tested different parameter choices for UMAP, with the clearest results generated by specifying 15 nearest neighbours (the default value) and a "minimum distance" between points in low dimensions of 0.5. UMAP developers described "sensible" values for nearest neighbours as between 5 and 50 and minimum distance between 0.5 and 0.001. Tuning these parameters will not change qualitative results much but may make patterns easier to identify. Increasing the number of neighbours will increase the computational load, and a smaller minimum distance can break the connectivity between clusters, though the same individuals will continue to group together.

UMAP and t-SNE projections were carried out on an iMac with a 3.5GhZ Intel Core i7 processor, 32 GB 1600 MHz DDR3 of RAM, and an NVIDIA GeForce GTX 775M 2048 MB graphics card.

Colours for maps in figures~\ref{fig:umap_world_map}, \ref{fig:umap_ukbb_geo_map}, \ref{fig:supp_umap_ukbb_asia}, \ref{fig:supp_umap_ukbb_car}, and \ref{fig:supp_umap_ukbb_eur} were determined by projecting data to 3D and using each 3D coordinate as an RGB coordinate. For the world map, countries were determined using the country of birth variable, with a country's colour being determined by the mean $x$, $y$, and $z$ values of all individuals born in that country. Because many self-identified White British individuals were born abroad, including them everywhere would skew the colour scheme; they were included only if they were born in the UK, Europe, Australia, Canada, or the United States. This approach to colouring is sensitive to sample sizes as UMAP will give more space to larger populations. For this reason, differences between Ireland/the UK and other countries appear exaggerated. Selecting a smaller minimum distance can result in breaking connections between clusters, leading to different colour groups appearing. We provide an example world map using 40 principal components and a minimum distance of 0.001 in figure~\ref{fig:supp_umap_ukbb_worldmap_alt}.

To reduce the potential risks for re-identification from results in this publication, data has been randomly permuted so that the population characteristics are preserved but individual-level data is not presented directly in the figures. We rounded each attribute to an attribute-specific number of bins, and then permuted the data in the following way: 
For each point (i.e. each individual) in UMAP visualizations, and each attribute, we identified the 9 nearest neighbouring points, and copied the attribute from a randomly selected neighbor (thus allowing for the possibility of one value being printed twice).  Because this process is done independently for each visualization, a given point shown on the figure will copy values from different randomly selected individuals. Additionally, spatial coordinates have random noise added (normally distributed about 0 with a standard deviation of 50km) before binning to the nearest 50km. For each point in figure~\ref{fig:umap_ukbb_geo_map} we identified the nearest 50 neighbouring individuals and copied the colour value from a randomly selected neighbour.}

\showmatmethods{} % Display the Materials and Methods section

%\acknow{}

%\showacknow{} % Display the acknowledgments section

\subsection*{Declarations}
\subsubsection*{Ethics approval and consent to participate}
HRS data was under IRB Study No. A11-E91-13B - The apportionment of genetic diversity within the United States. UKBB data was accessed under accession number 6728.

\subsubsection*{Consent for publication} Not applicable.

\subsubsection*{Availability of data and material} All data is publicly available to researchers.

\subsubsection*{Competing interests}
We have no competing interests to declare.

\subsubsection*{Funding}
This research was undertaken, in part, thanks to funding from the Canada Research Chairs program and CIHR grant MOP-136855.

\subsubsection*{Authors' contributions}
A.D.P. and S.G. designed the research. A.D.P. carried out dimension reduction and analysis. S.G. provided analysis. A.D.P and S.G. wrote the paper. L.A.T. provided the UK map visualization and analysis. C.B.E. prepared several UKBB datasets and principal component vectors.

\subsubsection*{Acknowledgements}
We thank all participants in the HRS, UKBB, and 1KGP for providing their genetic data as well as the teams who generated and assembled the datasets. We also thank Jose Sergio Hleap, Mark Lathrop, Dominic Nelson, Markus Munter, Stephen Sawcer, and Audrey Grant for useful discussions about science, programming, and data access; David Poznik, Liz Babalola, and Adam Auton from 23andMe for discussing findings in the 1KGP; and Selin Jessa for introducing us to UMAP. This research has been conducted using the UK Biobank Resource under Application Number 6728.

% Bibliography
\bibliography{pca_umap_genotype}

\newpage

\beginsupplement
\subsection*{Supporting Information (SI)}
\subsubsection*{SI Figures}
\begin{figure*}
    \centering
    \begin{subfigure}{\textwidth}
    \includegraphics[width=\textwidth]{images/megamontage_PC2_9.pdf}
    \end{subfigure}
    \caption{UMAP (left two columns) and t-SNE (right two columns) applied to the top principal components of the 1KGP labelled by the number of components used. Adding more components results in progressively finer population clusters using both methods.}
    \label{fig:supp_megamontage_pc2_9}  
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{0.95\textwidth}
    \includegraphics[width=0.95\textwidth]{images/megamontage_PC10_50.pdf}
    \end{subfigure}
    \caption{UMAP (left two columns) and t-SNE (right two columns) applied to the top principal components of the 1KGP labelled by the number of components used. Results are similar until approximately 11 components, where t-SNE breaks apart clusters of South Asian (in green) and Central and South American populations (in pink) while UMAP preserves them. At approximately 30 components populations begin to drift together with UMAP and disperse with t-SNE.}
    \label{fig:supp_megamontage_pc10_50}
\end{figure*}


\begin{figure*}
    \centering
    \begin{subfigure}{0.95\textwidth}
    \includegraphics[width=0.95\textwidth]{images/montage_1KGP_umap_convergence_resize.jpeg}
    \end{subfigure}
    \caption{UMAP applied to the first few hundred principal components of the 1KGP data. As more components are added, the figure begins to resemble that of UMAP carried out on the full genotype dataset.}
    \label{fig:supp_montage_1kgp_converge}
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{0.95\textwidth}
    \includegraphics[width=0.95\textwidth]{images/1KGP_UMAP_PCS100_PCE3450_NC2_NN15_MD05_201944185843.jpeg}
    \end{subfigure}
    \caption{UMAP applied the last 3350 principal components of the 1KGP.}
    \label{fig:supp_1kgp_3350}
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{\textwidth}
    \includegraphics[width=0.8\textwidth]{images/HRS_1000G_NP1_UMAP_PC7_NC2_NN15_MD05_pca_hrshisp_added1kgp_2018115153245_admix.pdf}
    \end{subfigure}
    \caption{UMAP of the first 7 principal components of the Hispanic population of the HRS, coloured by estimated admixture proportions.}
    \label{fig:supp_umap_hrs_hisp_admix}
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{0.95\textwidth}
    \includegraphics[width=0.75\textwidth]{images/HRS_1000G_NP1_UMAP_PC10_NC2_NN15_MD05_pca_1kgp_onto_hrs_umap_1kgp_onto_hrs_2018112221116_race_hisp_mex_labels.pdf}
    \end{subfigure}
    \caption{UMAP applied to the first 10 principal components of HRS data. Points coloured by self-identified race, Hispanic status, and Mexican-American status. The cluster on the left is mostly people who identify as neither Black nor White and were born outside the contiguous United States or in the Pacific census region. Clustering with the 1KGP data places them with Asian-identified populations. BNH, Black (not Hispanic); BHO, Black (Hispanic, Other); WNH, White (not Hispanic); WHM, White (Hispanic, Mexican-American); WHO, White Hispanic (Other); ONH, Other (not Hispanic); OHM, Other (Hispanic, Mexican-American); OHO, Other (Hispanic, Other).}
    \label{fig:supp_umap_hrs_eth}
\end{figure*}

\begin{figure}
\centering
   \includegraphics[width=0.6\linewidth]{images/HRS_1000G_NP1_UMAP_PC10_NC2_NN15_MD05_pca_1kgp_onto_hrs_umap_1kgp_onto_hrs_2018112221116_admix.pdf}
   \caption{UMAP on the first 10 principal components of HRS data. colouring individuals by estimated admixture from three ancestral populations reveals considerable diversity in the Hispanic population. This projection coloured by self-identified race and Hispanic status is presented in figure~\ref{fig:supp_umap_hrs_eth}.}
    \label{fig:umap_hrs_admix}
\end{figure}

\begin{figure*}
    \centering
    \begin{subfigure}{\textwidth}
    \includegraphics[width=0.75\textwidth]{images/HRS_1000G_NP1_UMAP_PC10_NC2_NN15_MD05_pca_1kgp_onto_hrs_umap_1kgp_onto_hrs_2018112221116_born.pdf}
    \end{subfigure}
    \caption{UMAP on the top 10 principal components of the HRS dataset, coloured by Census Bureau birth region. Each colour represents one of the 10 birth regions. There is no obvious pattern in the clusters of majority "White Not Hispanic" individuals.}
    \label{fig:supp_hrs_born}
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{\textwidth}
    \includegraphics[width=\textwidth]{images/HRS_1000G_UMAP_PC10_NC2_NN15_MD05_2018627203416_label.pdf}
    \end{subfigure}
    \caption{UMAP projection of the top 10 principal components of the combined HRS and 1KGP datasets. One cluster (in the box) does not group with any of the 1KGP populations. A cluster of Finnish (FIN) individuals consistently appears in the "White Not Hispanic" (WNH) group. Groups of Central and South American populations from the 1KGP (CLM, Colombian; MXL, Mexican; PEL, Peruvian; PUR, Puerto Rican) form nearby or within the HRS Hispanic cluster (HIS). Iberian individuals (IBS) cluster near the Hispanic population. Toscani individuals (TSI) form some small clusters and sometimes appear near the Iberian and Hispanic populations. Individuals with British/Scottish (GBR) or Northern/Western European ancestry (CEU) are scattered throughout the WNH clusters. Individuals with African ancestry from the 1KGP group with Black Americans from the HRS (AFR). Similar population groupings occur with South Asian (SAS) and East Asian (EAS) individuals.}
    \label{fig:supp_hrs_1000g}
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{\textwidth}
    \includegraphics[width=\textwidth]{images/HRS_1000G_NP1_UMAP_PC10_NC2_NN15_MD05_pca_1kgp_onto_hrs_umap_1kgp_onto_hrs_2018112221116_custom_label.pdf}
    \end{subfigure}
    \caption{UMAP on the top 10 principal components of the HRS data, with 1KGP data projected onto the embedding. Individuals from the HRS are grey. British (GBR) and other European (CEU) individuals are scattered throughout the "White Not Hispanic" clusters. Finns (FIN) form clear groupings. Spanish (IBS) and Italian (TSI) individuals cluster near the Hispanic grouping. There are sub-groups in the Hispanic cluster formed of Puerto Ricans (PUR), Colombians (CLM), Mexicans (MXL), and Peruvians (PEL). Populations with African ancestry (AFR) appear with Black individuals. East Asian (EAS) populations comprising Chinese, Kinh, and Japanese individuals cluster together with what appears in figure~\ref{fig:umap_hrs_admix} as a population of mostly Asian ancestry. South Asian (SAS) populations with Indian, Pakistani, and Sri Lankan ancestry cluster in a separate area. One "White Not Hispanic" cluster at the bottom does not cluster with any 1KGP populations.}
    \label{fig:supp_hrs_1kgp_projected}
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{\textwidth}
    \includegraphics[width=\textwidth]{images/HRS_PCGRID_8.pdf}
    \end{subfigure}
    \caption{Pairwise plots of the first 8 principal components of the Hispanic subset of the HRS. Those born in the Mountain region are coloured green.}
    \label{fig:supp_hrs_hisp_grid}
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{0.8\textwidth}
    \includegraphics[width=\textwidth]{images/UKBB_UMAP_PC10_NN15_MD05_eth_cob_resized.jpeg}
    \end{subfigure}
    \caption{Using country of birth data, some of the larger unidentified groups from figure~\ref{fig:supp_ukbb_zoom} were identified as being born mostly in Japan, the Philippines, North Africa, the Middle East, and Central and South America. The large cluster of "Any other Asian Background" were mostly born in Sri Lanka.}
    \label{fig:supp_ukbb_cob}
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{0.95\textwidth}
    \includegraphics[width=0.95\textwidth]{images/default_clean_size10_alpha60_flip.jpeg}
    \end{subfigure}
    \caption{UMAP on UKBB data, coloured by self-identified ethnic background. Images are labelled by the number of components included.}
    \label{fig:supp_montage_ukbb_eth}
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{0.95\textwidth}
    \includegraphics[width=0.95\textwidth]{images/default_clean_size10_alpha60_ns.jpeg}
    \end{subfigure}
    \caption{UMAP on UKBB data, coloured by northing values, with more blue representing more northern coordinates and more red representing more southern coordinates. Images are labelled by the number of components included. Data has been randomized as explained in the materials and methods section.}
    \label{fig:supp_montage_ukbb_ns}
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{\textwidth}
    \includegraphics[width=\textwidth]{images/default_clean_size10_alpha60_ew.jpeg}
    \end{subfigure}
    \caption{UMAP on UKBB data, coloured by easting values, with more yellow representing more eastern coordinates and more pink representing more western coordinates. Images are labelled by the number of components included. Data has been randomized as explained in the materials and methods section.}
    \label{fig:supp_montage_ukbb_ew}
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{\textwidth}
    \includegraphics[width=\textwidth]{images/meanAsiaMap.jpg}
    \end{subfigure}
    \caption{Figure~\ref{fig:umap_world_map}, zoomed in on Asia.}
    \label{fig:supp_umap_ukbb_asia}
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{\textwidth}
    \includegraphics[width=\textwidth]{images/meanCarribeanMap.jpg}
    \end{subfigure}
    \caption{Figure~\ref{fig:umap_world_map}, zoomed in on the Caribbean.}
    \label{fig:supp_umap_ukbb_car}
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{\textwidth}
    \includegraphics[width=\textwidth]{images/meanEuroMap.jpg}
    \end{subfigure}
    \caption{Figure~\ref{fig:umap_world_map}, zoomed in on Europe.}
    \label{fig:supp_umap_ukbb_eur}
\end{figure*}

\begin{figure}[!htb]
    \centering
    \includegraphics[width=0.95\columnwidth]{images/UKBB_TSNE_10PCs_DefaultPerplexity_eth.pdf}
    \caption{t-SNE applied to the top 10 principal components of the UKBB, coloured by ethnic background. The unbalanced populations resulted in many individuals and populations being orphaned along the periphery of the main cluster.}
    \label{fig:supp_ukbb_tsne}
\end{figure}

%%%%% UKBB phenotype plots
\begin{figure}
    \centering
    %\begin{subfigure}{0.8\textwidth}
    \includegraphics[width=0.4\columnwidth]{images/UKBB_UMAP_PC10_NN15_MD05_2018328174511_201871417039_basophill_count_pct5_f.pdf}
    %\end{subfigure}
    \caption{UMAP on the top 10 principal components of the UKBB coloured by basophil count (female). Data has been randomized as explained in the materials and methods section.}
    \label{fig:supp_ukbb_basophill_f}
\end{figure}

\begin{figure}
    \centering
    %\begin{subfigure}{0.8\textwidth}
    \includegraphics[width=0.4\columnwidth]{images/UKBB_UMAP_PC10_NN15_MD05_2018328174511_201871417039_basophill_count_pct5_m.pdf}
    %\end{subfigure}
    \caption{UMAP on the top 10 principal components of the UKBB coloured by basophil count (male). Data has been randomized as explained in the materials and methods section.}
    \label{fig:supp_ukbb_basophill_m}
\end{figure}

\begin{figure}
    \centering
    %\begin{subfigure}{0.8\textwidth}
    \includegraphics[width=0.4\columnwidth]{images/UKBB_UMAP_PC10_NN15_MD05_2018328174511_201871417720_eosinophill_count_pct5_f.pdf}
    %\end{subfigure}
    \caption{UMAP on the top 10 principal components of the UKBB coloured by eosinophil count (female). Data has been randomized as explained in the materials and methods section.}
    \label{fig:supp_ukbb_eosinophill_f}
\end{figure}

\begin{figure}
    \centering
    %\begin{subfigure}{0.8\textwidth}
    \includegraphics[width=0.4\columnwidth]{images/UKBB_UMAP_PC10_NN15_MD05_2018328174511_201871417720_eosinophill_count_pct5_m.pdf}
    %\end{subfigure}
    \caption{UMAP on the top 10 principal components of the UKBB coloured by eosinophil count (male). Data has been randomized as explained in the materials and methods section.}
    \label{fig:supp_ukbb_eosinophill_m}
\end{figure}

\begin{figure}
    \centering
    %\begin{subfigure}{0.8\textwidth}
    \includegraphics[width=0.4\columnwidth]{images/UKBB_UMAP_PC10_NN15_MD05_2018328174511_201871416305_3063_0_0_pct1_f.pdf}
    %\end{subfigure}
    \caption{UMAP on the top 10 principal components of the UKBB coloured by FEV1 (female). Data has been randomized as explained in the materials and methods section.}
    \label{fig:supp_ukbb_fev_f}
\end{figure}

\begin{figure}
    \centering
    %\begin{subfigure}{0.8\textwidth}
    \includegraphics[width=0.4\columnwidth]{images/UKBB_UMAP_PC10_NN15_MD05_2018328174511_201871416305_3063_0_0_pct1_m.pdf}
    %\end{subfigure}
    \caption{UMAP on the top 10 principal components of the UKBB coloured by FEV1 (male). Data has been randomized as explained in the materials and methods section.}
    \label{fig:supp_ukbb_fev_m}
\end{figure}

\begin{figure}
    \centering
    %\begin{subfigure}{0.8\textwidth}
    \includegraphics[width=0.4\columnwidth]{images/UKBB_UMAP_PC10_NN15_MD05_2018328174511_2018714161841_Height_res_pct1_f.pdf}
    %\end{subfigure}
    \caption{UMAP on the top 10 principal components of the UKBB coloured by height (female). Data has been randomized as explained in the materials and methods section.}
    \label{fig:supp_ukbb_height_f}
\end{figure}

\begin{figure}
    \centering
    %\begin{subfigure}{0.8\textwidth}
    \includegraphics[width=0.4\columnwidth]{images/UKBB_UMAP_PC10_NN15_MD05_2018328174511_2018714161841_Height_res_pct1_m.pdf}
    %\end{subfigure}
    \caption{UMAP on the top 10 principal components of the UKBB coloured by height (male). Data has been randomized as explained in the materials and methods section.}
    \label{fig:supp_ukbb_height_m}
\end{figure}

\begin{figure}
    \centering
    %\begin{subfigure}{0.8\textwidth}
    \includegraphics[width=0.4\columnwidth]{images/UKBB_UMAP_PC10_NN15_MD05_2018328174511_201871416519_leukocyte_count_pct5_f.pdf}
    %\end{subfigure}
    \caption{UMAP on the top 10 principal components of the UKBB coloured by leukocyte count (female). Data has been randomized as explained in the materials and methods section.}
    \label{fig:supp_ukbb_leukocyte_f}
\end{figure}

\begin{figure}
    \centering
    %\begin{subfigure}{0.8\textwidth}
    \includegraphics[width=0.4\columnwidth]{images/UKBB_UMAP_PC10_NN15_MD05_2018328174511_201871416519_leukocyte_count_pct5_m.pdf}
    %\end{subfigure}
    \caption{UMAP on the top 10 principal components of the UKBB coloured by leukocyte count (male). Data has been randomized as explained in the materials and methods section.}
    \label{fig:supp_ukbb_leukocyte_m}
\end{figure}

\begin{figure}
    \centering
    %\begin{subfigure}{0.8\textwidth}
    \includegraphics[width=0.4\columnwidth]{images/UKBB_UMAP_PC10_NN15_MD05_2018328174511_2018714165614_neutrophill_count_pct5_f.pdf}
    %\end{subfigure}
    \caption{UMAP on the top 10 principal components of the UKBB coloured by neutrophil count (female). Data has been randomized as explained in the materials and methods section.}
    \label{fig:supp_ukbb_neutrophill_f}
\end{figure}

\begin{figure}
    \centering
    %\begin{subfigure}{0.8\textwidth}
    \includegraphics[width=0.4\columnwidth]{images/UKBB_UMAP_PC10_NN15_MD05_2018328174511_2018714165614_neutrophill_count_pct5_m.pdf}
    %\end{subfigure}
    \caption{UMAP on the top 10 principal components of the UKBB coloured by neutrophil count (male). Data has been randomized as explained in the materials and methods section.}
    \label{fig:supp_ukbb_neutrophill_m}
\end{figure}

%%%%% UKBB phenotypes end

\begin{figure*}
    \centering
    %\begin{subfigure}{0.8\textwidth}
    \includegraphics[width=0.4\columnwidth]{images/ukbb_pcs_2019410184041_Height_res_pct1_f.jpeg}
    %\end{subfigure}
    \caption{Principal components 1 and 2 from the UKBB, coloured by age-adjusted residual height (female). Data has been randomized as explained in the materials and methods section.}
    \label{fig:supp_ukbb_pca_height_res_f}
\end{figure*}

\begin{figure*}
    \centering
    %\begin{subfigure}{0.8\textwidth}
    \includegraphics[width=0.4\columnwidth]{images/ukbb_pcs_201941019249_3063_0_0_pct1_f.jpeg}
    %\end{subfigure}
    \caption{Principal components 1 and 2 from the UKBB, coloured by FEV1 (female). Data has been randomized as explained in the materials and methods section.}
    \label{fig:supp_ukbb_pca_fev_f}
\end{figure*}

%\begin{figure*}
%    \centering
%    \includegraphics[width=0.4\columnwidth]{images/ukbb_pcs_201941019249_3063_0_0_pct1_f.jpeg}
%    \caption{Test. Data has been randomized as explained in the materials and methods section.}
%    \label{fig:supp_ukbb_tsne_fev_f}
%\end{figure*}

\clearpage
\begin{figure*}
    \centering
    %\begin{subfigure}{0.8\textwidth}
    \includegraphics[width=0.4\columnwidth]{images/UKBB_TSNE_10PCs_DefaultPerplexity_2019410184913_Height_res_pct1_f.jpeg}
    %\end{subfigure}
    \caption{t-SNE on the first 10 principal components from the UKBB, coloured by age-adjusted residual height (female). Data has been randomized as explained in the materials and methods section.}
    \label{fig:supp_ukbb_tsne_height_res_f}
\end{figure*}

\begin{figure*}
    \centering
    %\begin{subfigure}{0.8\textwidth}
    \includegraphics[width=0.4\columnwidth]{images/UKBB_TSNE_10PCs_DefaultPerplexity_2019410191255_3063_0_0_pct1_f.jpeg}
    %\end{subfigure}
    \caption{t-SNE on the first 10 principal components from the UKBB, coloured by FEV1 (female). Data has been randomized as explained in the materials and methods section.}
    \label{fig:supp_ukbb_tsne_fev_f}
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{0.8\textwidth}
    \includegraphics[width=\textwidth]{images/UKBB_UMAP_PC10_NN15_MD05_eth_combined_resized.pdf}
    \end{subfigure}
    \caption{Zoomed in areas of figure~\ref{fig:umap_ukbb_pc10}. Sections (i) and (ii) respectively focus on the African and Asian superpopulations, and section (iii) focuses on an area with individuals from many ethnic backgrounds. Noticeable clusters of unidentified ethnic backgrounds appear and are labelled "OEG" "(Other Ethnic Group)".}
    \label{fig:supp_ukbb_zoom}
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{\textwidth}
    \includegraphics[width=\textwidth]{images/montage_fev1_height_afr_permuted.pdf}
    \end{subfigure}
    \caption{Individuals of Black African, Black Caribbean, and mixed backgrounds (primarily White and Black Caribbean/African) coloured by self-identified ethnic background (left, from figure~\ref{fig:umap_ukbb_pc10}), FEV1 (middle), and age-adjusted height (right). An arrow points to an area where the FEV1 distribution appears to change, corresponding to where the clusters contain more people with self-identified mixed backgrounds.}
    \label{fig:supp_comparison_fev_afr}
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{\textwidth}
    \includegraphics[width=\textwidth]{images/montage_fev1_height_chi_eur_permuted.pdf}
    \end{subfigure}
    \caption{Zoomed in section of figure~\ref{fig:umap_ukbb_pc10} focused on individuals with Chinese (CHI), White British (GBR), any other white background, or any other ethnic group (OEG) coloured by ethnicity (left), FEV1 (middle), and age-adjusted height (right). The OEG cluster next to the Chinese cluster is coloured differently, suggesting this population may have different FEV1 characteristics. A cluster of OEG/other white individuals is more blue, suggesting they may have lower than average FEV1 values relative to the rest of the British or white population.}
    \label{fig:supp_comparison_fev_chi_eur}
\end{figure*}

\begin{figure}[!htb]
    \centering
    \includegraphics[width=0.95\columnwidth]{images/1KGP_tsne_umap.jpeg}
    \caption{Comparing the visualizations of UMAP, standard t-SNE, and t-SNE initialized with a UMAP projection, on the top 10 principal components of the 1KGP. t-SNE used 5000 iterations.}
    \label{fig:supp_tsne_umap_compare_1kgp}
\end{figure}

\begin{figure}[!htb]
    \centering
    \includegraphics[width=0.95\columnwidth]{images/HRS_tsne_umap.jpeg}
    \caption{Comparing the visualizations of UMAP, standard t-SNE, and t-SNE initialized with a UMAP projection, on the top 10 principal components of the HRS. t-SNE used 5000 iterations.}
    \label{fig:supp_tsne_umap_compare_hrs}
\end{figure}

\begin{figure}[!htb]
    \centering
    \includegraphics[width=0.95\columnwidth]{images/tsne_umap_graph_ukbb.jpeg}
    \caption{Comparing the error terms of standard t-SNE versus t-SNE initialized with a UMAP embedding and no early exaggeration. Done on the UKBB dataset with 20000 iterations. The UMAP-initialized graph has been shifted by 230 iterations to approximate the 230 epochs UMAP uses for large datasets ($n>10,000$).}
    \label{fig:supp_tsne_umap_compare_ukbb_graph}
\end{figure}

\begin{figure}[!htb]
    \centering
    \includegraphics[width=0.95\columnwidth]{images/ukbb_tsne_umap.jpeg}
    \caption{Comparing the visualizations of UMAP, standard t-SNE, and t-SNE initialized with a UMAP projection, on the top 10 principal components of the UKBB. t-SNE used 20000 iterations}
    \label{fig:supp_tsne_umap_compare_ukbb}
\end{figure}

\begin{figure}[!htb]
    \centering
    \includegraphics[width=0.95\columnwidth]{images/tsne_umap_graph_1kgp.jpeg}
    \caption{Comparing the error terms of standard t-SNE versus t-SNE initialized with a UMAP embedding and no early exaggeration. Done on the 1KGP dataset with 5000 iterations. The UMAP-initialized graph has been shifted by 600 iterations to approximate the 600 epochs UMAP uses for small datasets ($n\leq10,000$).}
    \label{fig:supp_tsne_umap_compare_1kgp_graph}
\end{figure}

\begin{figure}[!htb]
    \centering
    \includegraphics[width=0.95\columnwidth]{images/tsne_umap_graph_hrs.jpeg}
    \caption{Comparing the error terms of standard t-SNE versus t-SNE initialized with a UMAP embedding and no early exaggeration. Done on the HRS dataset with 5000 iterations. The UMAP-initialized graph has been shifted by 230 iterations to approximate the 230 epochs UMAP uses for large datasets ($n>10,000$).}
    \label{fig:supp_tsne_umap_compare_hrs_graph}
\end{figure}



%%%%% UKBB boxplots below

\begin{figure*}
    \centering
    \begin{subfigure}{\textwidth}
    \includegraphics[width=\textwidth]{images/female_basophill_boxplot_annotated.pdf}
    \end{subfigure}
    \caption{Boxplot of basophil counts by sex and ethnic group, annotated with p-values. Asterisks indicate significant difference from the White British group with a Bonferroni correction for 12 groups.}
    \label{fig:supp_box_basophill_f}
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{\textwidth}
    \includegraphics[width=\textwidth]{images/male_basophill_boxplot_annotated.pdf}
    \end{subfigure}
    \caption{Boxplot of basophil counts by sex and ethnic group, annotated with p-values. Asterisks indicate significant difference from the White British group with a Bonferroni correction for 12 groups.}
    \label{fig:supp_box_basophill_m}
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{\textwidth}
    \includegraphics[width=\textwidth]{images/female_eosinophill_boxplot_annotated.pdf}
    \end{subfigure}
    \caption{Boxplot of eosinophil counts by sex and ethnic group, annotated with p-values. Asterisks indicate significant difference from the White British group with a Bonferroni correction for 12 groups.}
    \label{fig:supp_box_eosinophill_f}
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{\textwidth}
    \includegraphics[width=\textwidth]{images/male_eosinophill_boxplot_annotated.pdf}
    \end{subfigure}
    \caption{Boxplot of eosinophil counts by sex and ethnic group, annotated with p-values. Asterisks indicate significant difference from the White British group with a Bonferroni correction for 12 groups.}
    \label{fig:supp_box_eosinophill_m}
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{\textwidth}
    \includegraphics[width=\textwidth]{images/female_fev_boxplot_annotated.pdf}
    \end{subfigure}
    \caption{Boxplot of FEV1 by sex and ethnic group, annotated with p-values. Asterisks indicate significant difference from the White British group with a Bonferroni correction for 12 groups.}
    \label{fig:supp_box_fev_f}
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{\textwidth}
    \includegraphics[width=\textwidth]{images/male_fev_boxplot_annotated.pdf}
    \end{subfigure}
    \caption{Boxplot of FEV1 by sex and ethnic group, annotated with p-values. Asterisks indicate significant difference from the White British group with a Bonferroni correction for 12 groups.}
    \label{fig:supp_box_fev_m}
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{\textwidth}
    \includegraphics[width=\textwidth]{images/female_height_boxplot_annotated.pdf}
    \end{subfigure}
    \caption{Boxplot of height by sex and ethnic group, annotated with p-values. Asterisks indicate significant difference from the White British group with a Bonferroni correction for 12 groups.}
    \label{fig:supp_box_height_f}
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{\textwidth}
    \includegraphics[width=\textwidth]{images/male_height_boxplot_annotated.pdf}
    \end{subfigure}
    \caption{Boxplot of height by sex and ethnic group, annotated with p-values. Asterisks indicate significant difference from the White British group with a Bonferroni correction for 12 groups.}
    \label{fig:supp_box_height_m}
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{\textwidth}
    \includegraphics[width=\textwidth]{images/female_leukocyte_boxplot_annotated.pdf}
    \end{subfigure}
    \caption{Boxplot of leukocyte counts by sex and ethnic group, annotated with p-values. Asterisks indicate significant difference from the White British group with a Bonferroni correction for 12 groups.}
    \label{fig:supp_box_leukocyte_f}
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{\textwidth}
    \includegraphics[width=\textwidth]{images/male_leukocyte_boxplot_annotated.pdf}
    \end{subfigure}
    \caption{Boxplot of leukocyte counts by sex and ethnic group, annotated with p-values. Asterisks indicate significant difference from the White British group with a Bonferroni correction for 12 groups.}
    \label{fig:supp_box_leukocyte_m}
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{\textwidth}
    \includegraphics[width=\textwidth]{images/female_neutrophil_boxplot_annotated.pdf}
    \end{subfigure}
    \caption{Boxplot of neutrophil counts by sex and ethnic group, annotated with p-values. Asterisks indicate significant difference from the White British group with a Bonferroni correction for 12 groups.}
    \label{fig:supp_box_neutrophill_f}
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{\textwidth}
    \includegraphics[width=\textwidth]{images/male_neutrophil_boxplot_annotated.pdf}
    \end{subfigure}
    \caption{Boxplot of neutrophil counts by sex and ethnic group, annotated with p-values. Asterisks indicate significant difference from the White British group with a Bonferroni correction for 12 groups.}
    \label{fig:supp_box_neutrophill_m}
\end{figure*}

%%%%% boxplots end
\begin{figure*}
    \centering
    \begin{subfigure}{\textwidth}
    \includegraphics[width=\textwidth]{images/meanWorldMap_mindist001.jpg}
    \end{subfigure}
    \caption{Alternate colouring of figre~\ref{fig:umap_world_map} using a UMAP projection on the top 40 components with minimum distance 0.001.}
    \label{fig:supp_umap_ukbb_worldmap_alt}
\end{figure*}


\begin{figure*}
    \centering
    \begin{subfigure}{\textwidth}
    \includegraphics[width=\textwidth]{images/admixture_plot_highlight_mountain_copy.pdf}
    \end{subfigure}
    \caption{Admixture plot of Hispanic individuals in the HRS. Those born in the Mountain census region fall between the white lines (indices 48 to 184)}
    \label{fig:supp_hrs_hisp_admix}
\end{figure*}

\end{document}