\documentclass[12pt]{pnas-new}
%\documentclass[draft]{pnas-new}
% Use the lineno option to display guide line numbers if required.
% Note that the use of elements such as single-column equations
% may affect the guide line number alignment. 

\newcommand{\beginsupplement}{%
        \setcounter{table}{0}
        \renewcommand{\thetable}{S\arabic{table}}%
        \setcounter{figure}{0}
        \renewcommand{\thefigure}{S\arabic{figure}}%
     }
\newcommand{\sgcomment}[1]{{\textcolor{blue}{SG: #1}}}
\newcommand{\adpcomment}[1]{{\textcolor{orange}{ADP: #1}}}

\newcommand{\rulesep}{\unskip\ \vrule\ }

%\usepackage{subfig}
\usepackage{adjustbox}
\usepackage{subcaption}
\usepackage{graphicx}
\usepackage{float}
\usepackage{dblfloatfix}
\usepackage[switch]{lineno}
\usepackage{comment}
%\usepackage[section]{placeins}

\linenumbers

\templatetype{pnasresearcharticle} % Choose template
\title{UMAP reveals cryptic population structure and phenotype heterogeneity in large genomic cohorts}

% Use letters for affiliations, numbers to show equal authorship (if applicable) and to indicate the corresponding author
\author[a,b]{Alex Diaz-Papkovich}
\author[b,c]{Luke Anderson-Trocm\'e}
\author[b,c]{Chief Ben-Eghan}
\author[b,c,1]{Simon Gravel} 
\affil[a]{Department of Quantitative Life Sciences, McGill University, Montreal, QC, H3A 0G1 Canada}
\affil[b]{McGill University and Genome Quebec Innovation Centre, Montreal, QC, H3A 0G1, Canada}
\affil[c]{Department of Human Genetics, McGill University, Montreal, QC, H3A 0G1, Canada. \textsuperscript{1}To whom correspondence should be addressed. E-mail: simon.gravel@mcgill.ca}

% Please include corresponding author, author contribution and author declaration information
\correspondingauthor{}

% Please give the surname of the lead author for the running footer
\leadauthor{Diaz-Papkovich} 

%An Analysis is a new analysis of existing data (typically large genomic, transcriptomic or proteomic data sets from arrays or other high-throughput platforms) or new data obtained in a comparative analysis of technologies that lead to novel and arresting conclusions of importance to a broad audience. The main text (excluding abstract, Methods, references and figure legends) is approximately 3,000 words. The abstract is typically 100-150 words, unreferenced. Analyses have no more than 6 display items (figures and/or tables). An introduction (without heading) is followed by sections headed Results, Discussion and online Methods. The Results and online Methods should be divided by topical subheadings; the Discussion does not contain subheadings. As a guideline, Analysis allow up to 50 references.

\begin{abstract}
{\large Abstract.}
%The abstract is typically 100-150 words, unreferenced. 
Human populations feature both discrete and continuous patterns of variation. Current analysis approaches struggle to jointly identify these patterns because of modelling assumptions, mathematical constraints, or numerical challenges. Here we apply uniform manifold approximation and projection (UMAP), a non-linear dimension reduction tool, to three well-studied genotype datasets and discover overlooked subpopulations within the American Hispanic population, fine-scale relationships between geography, genotypes, and phenotypes in the UK population, and cryptic structure in the Thousand Genomes Project data. This approach outperforms other visualization methods, is well-suited to the influx of large and diverse data, and opens new lines of inquiry in population-scale datasets.
\end{abstract}

\begin{document}

% Optional adjustment to line up main text (after abstract) of first page with line numbers, when using both lineno and twocolumn options.
\verticaladjustment{-2pt}

\maketitle
\thispagestyle{firststyle}
\ifthenelse{\boolean{shortarticle}}{\ifthenelse{\boolean{singlecolumn}}{\abscontentformatted}{\abscontent}}{}
Questions in medicine, anthropology, and related fields hinge on interpreting the deluge of genomic data provided by modern high-throughput sequencing technologies. Because genomic datasets are high-dimensional, their interpretation requires statistical methods that can comprehensively condense information in a manner that is understandable to researchers and minimizes the amount of data that is sacrificed. Both model-based and model-agnostic approaches to summarize data have played important roles in shaping our understanding of the evolution of our species~\cite{lawson2012inference} \sgcomment{This is a lone reference for a broad topic... Are there more recent reviews perhaps? Else add a couple of recent references eg by the novembre lab?}.

Here we will focus on nonparametric approaches to visualize relatedness patterns among individuals within populations. If we consider unphased single nucleotide polymorphism (SNP) data, an individual genome can be represented as a sequence of integers corresponding to the number of copies of the derived alleles carried by the individual at each of the $L$ SNPs for which genotypes are available, with $L$ ranging from hundreds of thousands to hundreds of millions. Since each individual is represented as an $L$-dimensional vector, dimension reduction methods are needed to visualize the data.

Principal component analysis (PCA) is often the first dimensional reduction tool used for genomic data. It identifies and ranks directions in genotype space that explain most-to-least variance among individuals. Positions of individuals along directions of highest variance can then be used to summarize individual genotypes. PCA coordinates have natural genealogical interpretations in terms of expected times to a most recent common ancestor (TMRCA) \cite{mcvean2009genealogical}, and are used empirically to reveal admixture \cite{brisbin2012pcadmix}, continuous isolation-by-distance \cite{novembre2008europe, nelson2008population}, as well as technical artefacts. PCA coordinates are particularly well-suited to correct for population structure in GWAS~\cite{eigen2006}.

\begin{figure*}[!ht]
    \centering
    \includegraphics[width=0.7\textwidth]{images/montage_1000G_labels_axes.pdf}
    \caption{\textbf{Four methods of dimension reduction of 1KGP genotype data with population labels.} (i) PCA maps individuals in a triangle with vertices corresponding to African, Asian, and European continental ancestry. Discarding lower-variance PCs leads to overlap of populations with no close affinity, such as Central and South American populations with South Asians. (ii) t-SNE forms groups corresponding to continents, with some overlap between European and Central and South American people. Smaller subgroups are visible within continental clusters. The cloud of peripheral points results from the method's poor convergence. (iii) UMAP forms distinct clusters related to continent with clearly defined subgroups. Japanese, Finnish, Luhya, and some Punjabi and Telugu populations form separate clusters consistent with their population history\cite{10002015global}. (iv) UMAP on the first 15 principal components forms fine-scale clusters for individual populations. Groups closely related by ancestry or geography, such as African Caribbean/African American, Spanish/Italian, and Kinh/Dai populations cluster together. Results using t-SNE on principal components are presented in figure~\ref{fig:supp_megamontage_pc2_9}. Axes in UMAP and t-SNE are arbitrary. Since the algorithms prioritize local distances, long distances between clusters are not meaningful.
    ACB, African Caribbean in Barbados;
    ASW, African Ancestry in Southwest US;
    BEB, Bengali;
    CDX, Chinese Dai;
    CEU, Utah residents with Northern/Western European ancestry;
    CHB, Han Chinese;
    CHS, Southern Han Chinese;
    CLM, Colombian in Medellin, Colombia;
    ESN, Esan in Nigeria;
    FIN, Finnish;
    GBR, British in England and Scotland;
    GWD, Gambian;
    GTH, Gujarati;
    IBS, Iberian in Spain;
    ITU, Indian Telugu in the UK;
    JPT, Japanese;
    KHV, Kinh in Vietnam;
    LWK, Luhya in Kenya;
    MSL, Mende in Sierra Leone;
    MXL, Mexican in Los Angeles, California;
    PEL, Peruvian;
    PJL, Punjabi in Lahore, Pakistan;
    PUR, Puerto Rican;
    STU, Sri Lankan Tamil in the UK;
    TSI, Tuscani in Italy;
    YRI, Yoruba in Nigeria}
    \label{fig:1000g_compare}
\end{figure*}

The amount of information encoded in the lower-variance principal components increases with sample size, so researchers typically examine multiple two-dimensional projections to explore data. In this process, finer features of the data may be hidden by the projections or hard to interpret. To display finer features of the data in a two dimensional figure, we can use non-linear transformations that emphasize the local structure of the data. A popular method for such visualization is t-distributed stochastic neighbour embedding (t-SNE)\cite{maaten2008visualizing}. t-SNE has been used before to visualize SNPs\cite{platzer2013visualization}; using data from the 1000 Genomes Project (1KGP)\cite{10002015global}, it groups individuals corresponding roughly to their continent of origin, with smaller ethnic sub-groups visible within the larger continental clusters\cite{li2017tsne}. However, t-SNE struggles with very large datasets, when a large number of locally optimal configurations make convergence to a globally satisfying solution difficult. 

Uniform Manifold Approximation and Projection (UMAP) is a dimension reduction technique designed to model and preserve the high-dimensional topology of data points in the low-dimensional space\cite{2018arXivUMAP}. With genotype data, UMAP creates a neighbourhood around each individual's genetic coordinates and identifies a pre-selected number of neighbours to build high-dimensional manifolds. The end result is a patchwork of low-dimensional representations of neighbourhoods that groups genetically similar individuals together on a local scale while better preserving long-range topological connections to more distantly related individuals. The method has been successfully applied to single-cell RNA sequencing datasets\cite{umap2018singlecell}.

Nonlinear dimension reduction methods tend to be computationally intensive. A common practice to reduce this burden is to first apply PCA to data, and perform dimensional reduction on data projected to leading principal components (PCs). In addition to being computationally advantageous, this discards noise that can confound nonlinear approaches: population structure arising from $n$ isolated randomly-mating demes can be described by the leading $n-1$ PCs, with the following PCs describing stochastic variation in relatedness~\cite{eigen2006}. Selecting the leading PCs therefore has potential to extract meaningful population structure while filtering out stochastic noise. We explore different strategies to pre-process the data and investigate discrete and continuous population structure patterns present in large datasets of human genotypes: the 1KGP, the Health and Retirement Study (HRS)\cite{juster1995overview}, and the UK BioBank (UKBB)\cite{sudlow2015uk}, and compare UMAP's performance to t-SNE.  

\begin{figure*}
    \centering
    \begin{subfigure}{0.49\columnwidth}
    \includegraphics[width=\columnwidth]{images/HRS_1000G_NP1_UMAP_PC7_NC2_NN15_MD05_pca_hrshisp_added1kgp_2018115153245_1kgp_hisp.jpeg}
    \caption{}
    \label{fig:supp_umap_hrs_hisp_1kgp_hisp}
    \end{subfigure}\hfill
    \unskip\ \vrule\
        \begin{subfigure}{0.49\columnwidth}
    \includegraphics[width=\columnwidth]{images/Asthma_FullAsian_PC_QC_new_UMAP_PC5_NC2_NN15_MD05_201931173256_ethcob.jpeg}
    \caption{}
        \label{fig:umap_ukbb_indian}
    \end{subfigure}
    \caption{\textbf{Applying UMAP to subsets of data can reveal deep population structure.} \textbf{(a)} UMAP on the top $7$ principal components of the self-identified Hispanic population of the HRS reveals a cluster. Colouring the points by birthplace shows they were born almost entirely in the Mountain region (in green) of the United States (New Mexico, Arizona, Colorado, Utah, Nevada, Wyoming, Idaho, and Montana). When populations from the 1KGP are projected onto the UMAP embedding they do not map to the cluster. Six 1KGP populations are presented:
    CLM, Colombian in Medellin, Colombia;
    IBS, Iberian in Spain;
    MXL, Mexican in Los Angeles, California;
    PEL, Peruvian;
    PUR, Puerto Rican;
    TSI, Tuscani in Italy. Figures~\ref{fig:supp_umap_hrs_hisp_admix} and \ref{fig:supp_umap_hrs_hisp_birth} present the same projection of individuals from the HRS coloured by estimated admixture proportions census region of birth, respectively. \textbf{(b)} UMAP on the top $8$ principal components of the self-identified Asian populations of the UKBB create clusters. Indian individuals born in Kenya (in purple) form one such cluster. A version coloured by self-identified ethnicity is presented in figure~\ref{fig:supp_umap_ukbb_asian_eth}. }
    \label{fig:subpops}
\end{figure*}

\subsection*{Results}
\subsubsection*{Fine-scale visualization of the 1KGP dataset} The 1KGP contains genotype data of 3,450 individuals from 26 relatively distinct labeled populations\cite{10002015global}. Figure~\ref{fig:1000g_compare} shows visualizations using PCA, t-SNE, UMAP, and UMAP with PCA pre-processing. Using UMAP and t-SNE on the genotype data presents clusters that are roughly grouped by continent, with UMAP showing a clear hierarchy of population and continental clusters, whereas t-SNE fails to assign many individuals to population clusters. Using either method on the top principal components leads to distinct population clusters and less defined continental structure. Adding more components results in progressively finer clusters until approximately 20 populations appear using 15 components; adding further components converges to results similar to using the entire genotype data (see figures~\ref{fig:supp_megamontage_pc2_9}, \ref{fig:supp_megamontage_pc10_50}, and \ref{fig:supp_montage_1kgp_converge}). To investigate the population information contained in low-variance PCs, we performed UMAP on data projected onto PCs 100 to 3450 (i.e., without information about the leading 99 PCs). Figure~\ref{fig:supp_1kgp_3350} shows that population structure is still clearly visible. \sgcomment{Note figure S4 lacks a color code -- at least point out that it uses the same coloring as X} 

Focusing on UMAP with the leading 15 principal components (figure~\ref{fig:1000g_compare}~(iv)), several population clusters reflect shared ancestries. British individuals from England and Scotland form a cluster mixed with those from Utah who claim Northern and Western European ancestry. Toscani and Iberian individuals form a cluster reflecting their Mediterranean heritage. African Americans in the Southwest US, African Caribbean individuals in Barbados, and some Puerto Ricans also form a cluster. Three East Asian clusters appear: one is largely Han and Southern Han individuals, another is comprised of the Chinese Dai in southern China and the Kinh from Vietnam, and the third is the Japanese population. Other clusters are comprised of Colombians and Peruvians, the Esan and Yoruba populations of Nigeria, and several South Asian populations. 

Within population clusters, family members were projected near each other within broader population groups. When UMAP was parameterized to use only 5 nearest neighbours, however, families often formed distinct clusters (figure~\ref{fig:supp_1kgp_families}): Using few neighbors to build a manifold emphasizes closer relatedness.

A few individuals cluster with populations different from their label: some Mexican individuals cluster with Spanish and Italian populations; some Puerto Rican individuals cluster with African American and Caribbean populations; and one Gambian individual clusters with the Mende of Sierra Leone. Two populations form multiple clusters: Gujarati Indians in Houston, Texas and Punjabi people in Lahore, Pakistan. This clustering effect is robust to the number of components considered (figure~\ref{fig:supp_megamontage_pc10_50}). Differentiation in the Gujarati population from substructure has been previously noted\cite{reich2009india}. Recontacting of 23andMe participants associated with the clusters show that one of the cluster is highly enriched in individuals from a group in Western India with shared ancestry and patronym. [D. Poznik, 23andme, personal communication, and \cite{23andme}].

\subsubsection*{Admixed individuals fall along a genetic continuum}
The 1KGP sampled individuals from relatively distinct populations, so the data are more likely to form clusters. Most medical cohorts, however, comprise larger numbers of individuals sampled across extended geographical areas. The HRS contains genotype data of 12,454 Americans from a variety of backgrounds. Using UMAP on the first 10 principal components, we demonstrate projections that present a collection of sub-populations and a continuum of genetic variation.

The HRS forms several large clusters, reflecting both ethnicity (figure~\ref{fig:supp_umap_hrs_eth}) and admixture proportions (figure~\ref{fig:umap_hrs_admix}). Gradients in admixture proportion are visible within the predominantly Hispanic cluster, but not within the predominantly Black cluster, perhaps because the variance in ancestry proportions is greater among Hispanics. The "White Not Hispanic" (WNH) group forms several interconnected clusters, and these do not correspond to broad geographical areas (figure~\ref{fig:supp_hrs_born}). By generating the PC axes and UMAP embedding for the HRS data in figure~\ref{fig:supp_umap_hrs_eth}, and projecting the 1KGP data onto it, we reveal substructure within the Hispanic cluster, groupings of Finnish individuals within the WNH groups, as well as Italian and Spanish individuals grouping near the White Hispanic population (figure~\ref{fig:supp_hrs_1kgp_projected}). One group of WNH individuals regularly appears at the periphery of the main cluster and does not cluster with any 1KGP populations.

\subsubsection*{Regional patterns in the Hispanic subpopulation} Applying UMAP to self-identified Hispanic individuals in the HRS reveals clear groupings related to birth region. One cluster, highlighted in figure~\ref{fig:supp_umap_hrs_hisp_1kgp_hisp}, consists almost entirely of individuals born in the Mountain Region of the United States. This cluster is not apparent when looking at a grid of pairwise plots of the first 8 principal components, provided in figure~\ref{fig:supp_hrs_hisp_grid}, as the signal is distributed along PCs 3, 4, and 6. Even though continental admixture patterns do correlate with UMAP position (figure~\ref{fig:supp_umap_hrs_hisp_admix}), these do not explain the Mountain Region cluster. Individuals from 1KGP populations do not appear in the cluster when projected to the UMAP embedding. The cluster possibly comprises the Hispano/Nuevomexicano population of the Southwest US, who have been present in the Mountain Region area long before the more recent immigrants from Latin America, and whose ancestry is expected to reflect both distinct Native ancestry and population-specific drift relative to other Hispanic populations. Such a cluster has been previously identified in AncestryDNA data using sequential clustering on identity-by-descent connections\cite{han2017clustering}; a recent preprint discusses the Mountain Region Hispanics with a more detailed historical description\cite{Jordan333609}.  

\subsubsection*{Population structure in the UKBB reflects local and global genetic variation} 
The UKBB contains data on 488,377 individuals containing phenotypic measures and self-identified ethnic backgrounds. Figure~\ref{fig:fig_ukbb} compares UMAP to PCA apply to UKBB. As expected, PCA captures major axes of variation emphasizing continental ancestry, whereas UMAP reveals finer sturcture. UMAP on the top 10 principal components reveals continuous and discrete population structure (figure~\ref{fig:umap_ukbb_pc10}): the patchwork of local topologies identifies multiple sub-populations, as well as continuous structure within populations and admixture gradients between populations. The result is a comprehensive portrait of genetic variation capturing population relationships not visible using other methods, succinctly illustrating the complex structure of a large and multi-ethnic dataset.

\begin{figure*}%
\centering
\begin{subfigure}{.5\columnwidth}
\includegraphics[width=\columnwidth]{images/UKBB_PC0_PC1_eth_crop.jpeg}%
\caption{}%
\label{fig:pc_ukbb}%
\end{subfigure}\hfill%
\begin{subfigure}{.5\columnwidth}
\includegraphics[width=\columnwidth]{images/UKBB_UMAP_PC10_NN15_MD05_eth_labels.pdf}%
\caption{}%
\label{fig:umap_ukbb_pc10}%
\end{subfigure}%
\caption{\textbf{The UKBB coloured by self-reported ethnic background.} \textbf{(a)} The first two principal components, showing the usual triangle with vertices corresponding to African, Asian, and European ancestries, and intermediate values indicating admixture or lack of relationship to the vertex populations. \textbf{(b)} UMAP on the first 10 principal components. The cluster of White British and White Irish individuals is greatly expanded, with the Irish forming a distinct sub cluster mixed with the White British population. South Asian and East Asian individuals form their separate clusters, as do individuals of African or Caribbean backgrounds. Population clusters are connected by "trails" comprised of large proportions of individuals with mixed backgrounds.
BA, Black African; 
BC, Black Caribbean; 
BG, Bangladeshi; 
CHN, Chinese; 
IND, Indian;
PK, Pakistani;
WB, White British;
WI, White Irish;
WBC, White and Black Caribbean; 
WBA, White and Black African; 
WAA, White and Asian;
AAB, Any other Asian Background; 
ABB, Any other Black Background;
AWB, Any other White Background;
AMB, Any other Mixed Background;
OEG, Other ethnic group. 
}
\label{fig:fig_ukbb}
\end{figure*}

The largest cluster in figure~\ref{fig:umap_ukbb_pc10} consists of the White British and Irish populations. The Irish population forms a sub-cluster, but many individuals are also scattered throughout the British-identifying population. Individuals identifying as Black African and Black Caribbean partially overlap, but admixed individuals form distinct trails leading to Asian and European clusters. Chinese individuals form a cluster, within a broader East Asian population; Indian, Pakistani, and Bangladeshi populations form a closely bound cluster as well. The East Asian and South Asian populations each have large clusters of individuals who identify as having an "other Asian background" or belonging to an "other ethnic group". The patchwork of genetic neighbourhoods is connected by trails of admixed individuals, which converge at a nexus of individuals with a variety of ethnicities. Many claim mixed ancestry, and there are clusters of individuals who belong to an "other ethnic group". Using data on countries of birth, we identified many finer groups in figure~\ref{fig:supp_ukbb_cob}, and confirmed they appeared in intuitive areas with, e.g., Japanese and Filipino clusters being projected near Chinese clusters.

Figure~\ref{fig:umap_ukbb_geo_osgb} presents the UMAP projection from figure~\ref{fig:umap_ukbb_pc10} coloured instead by geographical coordinates from the Ordnance Survey National Grid (OSGB1936), with distances defined as a north or east position relative to the Isles of Scilly. Geographic clusters form in the large White British grouping, reflecting the relationship between genetic and geographic distance, as has been observed in Europe and British-wide data\cite{novembre2008europe,leslie2015fine}. Figure~\ref{fig:umap_ukbb_pc10} shows that the admixed individuals have UMAP coordinates next to White British individuals residing in the South East of the UK, where London is located (see also figure~\ref{fig:supp_london_distance}, where individuals are coloured by distance from London). This likely reflects the high migration levels to the city and surrounding area: the UMAP projection attempted to preserve both the genetic similarity among admixed individuals and the relatedness with White British individuals in cosmopolitan areas. 

 The detailed shape of extended clusters is not stable as we vary the number of PCs included, but the patterns mentioned above are preserved. Figures~\ref{fig:supp_montage_ukbb_eth}, \ref{fig:supp_montage_ukbb_ns}, and~\ref{fig:supp_montage_ukbb_ew} show UMAP plots using the top $40$ PCs from the UKBB.

\begin{figure}
    \centering
    \begin{subfigure}{\columnwidth}
    \includegraphics[width=0.4\columnwidth]{images/UKBB_UMAP_PC10_NN15_MD05_2018328174511_ns_permuted_10nn_sd50000_base50000_201883118582.pdf}
    \includegraphics[width=.4\columnwidth]{images/UKBB_UMAP_PC10_NN15_MD05_2018328174511_ew_permuted_10nn_sd50000_base50000_201883118582.pdf}
    \caption{}
    \label{fig:umap_ukbb_geo_osgb}
    \end{subfigure}
     \begin{subfigure}{\columnwidth}
         \includegraphics[width=0.4\columnwidth]{images/UKBB_UMAP_PC10_NN15_MD05_2018328174511_2018714161841_Height_res_pct1_f.pdf}
    \includegraphics[width=.4\columnwidth]{images/UKBB_UMAP_PC10_NN15_MD05_2018328174511_201871416519_leukocyte_count_pct5_f.pdf}
   \caption{}
    \label{fig:umap_height_female}
      \end{subfigure}
      \caption{\textbf{UMAP captures relationships between population structure, geography, and phenotype heterogeneity.} \textbf{(a)} Each individual is coloured by their geographical coordinates of residence. Coordinates follow the UKBB's OSGB1936 geographic grid system and represent distance from the Isles of Scilly, which lie southwest of Great Britain. The left image colours individuals by their north-south ("northing") coordinates, and the right image colours them by their east-west ("easting") coordinates. Adding more components creates finer clusters.  (figures~\ref{fig:supp_montage_ukbb_ns} and~\ref{fig:supp_montage_ukbb_ew}). Northing values were truncated between $100$km and $700$km, and easting values were truncated between $200$km and $600$km. \textbf{(b)} Females from the same projection coloured by age-adjusted difference from mean population height (left) and leukocyte counts (right). Individuals with missing data were excluded. To protect participant privacy, data in these images has been randomized as explained in the materials and methods section.}
\end{figure}

As an alternate visualization of geography and genetic diversity, we performed a 3D UMAP projection and converted the UMAP coordinates into RGB values, allowing us to plot individuals on a map of Great Britain, emphasizing both spatial gradients of genetic relatedness and increased diversity in urban centers (figure~\ref{fig:umap_ukbb_geo_map}). The geographical patterns outside major urban centers are similar to those reported in~\cite{leslie2015fine} using the haplotype-based CHROMOPAINTER on British individuals whose grandparents lived nearby. Using data about country of birth, we performed a similar analysis of a world map~\ref{fig:umap_world_map}, revealing subtle regional variation around the world. 

\begin{figure}[h!]
  \centering
  \begin{subfigure}[b]{0.4\linewidth}
    \includegraphics[width=\linewidth]{images/UKBB_mapsInMaps_2D3_permute50.jpg}
    \caption{}
    \label{fig:umap_ukbb_geo_map}
  \end{subfigure}
  \begin{subfigure}[b]{0.4\linewidth}
    \includegraphics[width=\linewidth]{images/meanWorldMap.jpg}
    \caption{}
  \label{fig:umap_world_map}
  \end{subfigure}
  \caption{\textbf{Maps coloured by 3D UMAP projections of the top 20 principal components of the UKBB.} Each individual is assigned a 3D RGB vector based on 3D UMAP coordinates (a flattened projection is in the top right of \textbf{(a)}. Individuals who are closer to each other in the projection will be closer in colour in the maps. More details on colouring, as well as randomization of points to protect participant privacy, are available in the materials and methods. \textbf{(a)} Each point is an individual placed based on where they live. Patterns in genetic similarity are visible in Scotland, South England, North and South Wales, the East and West Midlands, and major urban centres. \textbf{(b)} Geographic distribution of UMAP coordinates. Using the country of birth of individuals in the UKBB, we colour countries by the closeness in 3D UMAP space of those born there. Broad patterns of similarity appear in East Asia, South Asia, North African and the Middle East, West Africa, and South America. Differences between neighbouring countries can reflect both ancient population structure and recent differences in migration history. Evidence of migrations related to colonialism are visible with, e.g., European ancestry in South Africa and South Asian ancestry in Kenya and Tanzania. Because of the large number of White British individuals born abroad, to avoid skewing the colour scale they were not included unless they were born in the UK, Europe, Australia, Canada, or the United States, where UKBB participants already tended to have European ancestry. Zoomed maps of East Asia, the Caribbean, and Europe are available in figures~\ref{fig:supp_umap_ukbb_asia}, \ref{fig:supp_umap_ukbb_car}, and \ref{fig:supp_umap_ukbb_eur}, respectively.}
\end{figure}

Similarly to UMAP, t-SNE applied to the UKBB data both displays diversity within the "White British" population and identifies clusters among other groups. However, it has three drawbacks: it is much slower, requiring 2.26 hours for its first thousand iterations alone on 10 principal components against UMAP's 14 minutes; it fails to find a global optimum, which results in a scattering of individuals and groups that are not stable across independent runs; and it does not identify continuity between different continental groups resulting from admixture (figure~\ref{fig:supp_ukbb_tsne}). 

\subsubsection*{Identifying patterns in phenotype variation related to genetic population structure} 

Covariates such as height and leukocyte count (figure~\ref{fig:umap_height_female}) and autoimmune and asthma-related measures (figures~\ref{fig:supp_ukbb_basophill_f} to \ref{fig:supp_ukbb_neutrophill_m}) correlate strongly with both discrete and continuous population structure. Several populations in figure~\ref{fig:umap_height_female}, including South Asian, East Asian, African, and others have noticeably lower-than-average heights. More subtle patterns are also visible: the area of the projection in figures~\ref{fig:umap_ukbb_pc10} with the cluster of White Irish people appears more blue than the main body of White British individuals. To quantify and statistically test these qualitative observations, we performed an unpaired two sample t-test of self-identified White Irish and White British individuals and found British males taller on average by 0.846cm (p-value $2.10\times 10^{-23}$) and British females by 0.763cm (p-value $3.65\times 10^{-23}$) (see figures~\ref{fig:supp_box_height_f} and \ref{fig:supp_box_height_m} for boxplots). Height differences between Irish and British populations have been previously observed but the direction of the difference is not consistent\cite{robinson2015population,komlos1994stature}.

Forced expiratory volume in 1 second (FEV1) (figures~\ref{fig:supp_ukbb_fev_f}, \ref{fig:supp_ukbb_fev_m}) also shows strong correlations with certain populations --- South Asian, African, and Caribbean --- having considerably lower measurements on average (see figures~\ref{fig:supp_box_fev_f} and \ref{fig:supp_box_fev_m} for boxplots and p-values). Notably, there appears to be a juncture in the admixture population, highlighted in figure~\ref{fig:supp_comparison_fev_afr}, where the distribution of FEV1 changes. This roughly corresponds to the transition from Black African/Caribbean individuals to those who identified having mixed backgrounds. Boxplots and statistical testing suggest that relative to White British populations, FEV1 values are significantly lower for Black African and Black Caribbean populations, but not for White and Black Caribbean and White and Black African populations (figures~\ref{fig:supp_box_fev_f} and \ref{fig:supp_box_fev_m}). 

\adpcomment{Discussing FEV1 here. Note: citations tend to present relative differences b/w groups as a percentage.}
Figure~\ref{fig:supp_comparison_fev_chi_eur} further suggests a difference in FEV1 between those who self-identified as Chinese and a nearby cluster enriched in individuals born in Japan. Baseline FEV1 levels are determined via equations that include ancestry or race\cite{quanjer2012multi}, but studies in several populations have shown that there is considerable variation based on ancestry, even within self-defined ethnicity\cite{ortega2015effect}; we are not aware of any results comparing populations within Asia. We selected the individuals whose UMAP coordinates were near the Chinese cluster and kept those born in Japan, Malaysia, and the Philippines (the largest populations in the UKBB by country of birth) as well as the self-identified Chinese population (mutually exclusive groups, shown in figure~\ref{fig:supp_fev_test_pops}). After adjusting for age, age$^2$, height, and sex, an unpaired two-sample t-test shows those born in Japan have a higher mean FEV1 than Chinese individuals by 0.224 (p-value $2.787\times 10^{-15}$). By sex, there is a difference of 0.213 (p-value $5.4\times 10^{-13}$) among females and 0.317 (p-value $5.1\times 10^{-4}$) among males, though there are considerably fewer males in the sample (distributions presented in figure~\ref{fig:supp_fev_ridgeplots}). For comparison, the adjusted difference between self-identified African and British individuals in the UKBB is 0.762 (p-value $2.2\times 10^{-16}$).

%For reference, \sgcomment{US organisation} recommends using different FEV1 reference scales for asthma diagnostic in European-Americans and African-Americans based on a population mean difference in FEV1 of XX.  

%Some of the phenotypic variation patterns mentioned above could, in principle, have been identified by considering large numbers of box plots, but we find that UMAP facilitates the identification of large phenotypic differences among genetically related groups. This is especially useful when population labels based on self-reported identity or country of birth are unavailable or correlate poorly with genetic ancestry.

\subsubsection*{Comparing t-SNE and UMAP}
\sgcomment{tweaked this a bit -- would be good to include in reply to reviewer}
Identifying the best dimension reduction technique is challenging, both because the `best' representation depends on context, and because convergence issues may mean that a good theoretical model for dimensional reduction might perform poorly because of challenges in numerical optimization. To assess whether the relatively poor performance of t-SNE could be due to convergence rather than a flawed model, we used UMAP to preprocess the UKBB data and provide a starting point to a standard t-SNE implementation. This led to representations that were objectively better (according to the t-SNE metric) than the default t-SNE implementation (figure~\ref{fig:supp_tsne_umap_compare_ukbb_graph}). Yet, these representations were much less detailed than the UMAP embedding provided as a starting point (figure~\ref{fig:supp_tsne_umap_compare_ukbb}).  Given these results, we recommend UMAP over t-SNE for large and diverse genomic datasets.

\subsection*{Discussion}
UMAP comprehensively illustrates genotypic information at fine scales and within the context of global population structure. It is easy to use and fast; given PCA data and a desktop computer, UMAP can be performed in 15 to 25 minutes on a sample of hundreds of thousands of individuals over tens of dimensions. It excels with larger datasets containing individuals with admixed backgrounds, which present discrete and continuous population structure. By patching together many local genetic neighbourhoods, the visualizations preserve aspects of both local and global structure, unlike any existing method that we are familiar with. 

Using UMAP reveals  clusters that would have been difficult to identify via pairwise PCA plots or Admixture analysis, such as the geographically restricted cluster within the Hispanic population of the HRS, or the splits within the Gujarati and Punjabi population samples in the 1KGP. More importantly, UMAP helps reveal patterns of covariation between geography, phenotypes and genotypes. Traits such as height showed continuous variation across admixture edges and geographic gradients, as expected from genetically controlled complex traits, and others, such as leukocyte counts or FEV1, showed sharper boundaries and nonlinear behavior consistent with the existence of strong regional environmental influences. 

We found that pre-processing the data with PCA allowed for time savings, but identifying an optimal number of PCs to use is challenging. Groupings on ethnicity formed slowly as PCs were added until reaching a stable number around 10 to 15 PCs. Geographical patterns in the UKBB continued to appear even up to 40 components, as visible in figures~\ref{fig:supp_montage_ukbb_ns} and \ref{fig:supp_montage_ukbb_ew}. 

\subsubsection*{Caveats}
The existence of a (small) number of adjustable parameter to UMAP, including PC cutoff and number of neighbors, opens the door to tweaking. Figure~\ref{fig:supp_megamontage_pc10_50} shows that disconnected population clusters can appear at different locations depending on parameter choices. In the absence of clear theoretical rationales, we generally recommend to use as many PCs as are available and computationally feasible, and to use the default number of neighbours (15). We sometimes found that lower number of PCs led to a simpler shape that facilitated discussion (e.g., figure \ref{fig:fig_ukbb}). We recommend reporting on a range of parameter values and following up on observations with statistical testing. Like most non-linear methods, UMAP lacks direct interpretability. It emphasizes local distances over global distances; while points that are very close in UMAP space are likely close in the original data, points that are distant in UMAP space are not necessarily very different in the original data. For these reasons, UMAP coordinates should not be used as GWAS covariates or for quantifying distances between populations. 

UMAP is sensitive to sample sizes and spends more visual real estate on populations with larger sample sizes. This is useful to identify significant patterns in a cohort, but it makes comparing visualization across cohorts difficult and may appear to exaggerate the genetic variation within the most sampled populations, such as the White British population in the UKBB. We did not assign meaning to wiggles in UMAP figures, which occurred consistently in the UKBB but may be an artefact of the dimensional reduction strategy rather than a meaningful feature of the data. Hand-waving interpretations of pretty plots has a history of getting population geneticists in trouble (as pointed out, e.g., in \cite{novembre2008interpreting}): visualization is not a replacement for statistical testing.

With these caveats in mind, a priori data visualization plays a central role in quality control, hypothesis generation, and confounder identification for a wide range of genomic applications. Nonlinear approaches, despite their limitations, become increasingly useful as the size of datasets increases. UMAP, in particular, reveals a wide range of features that would not be apparent using linear maps. Given its ease of use, breadth of results, and low computational cost, we propose that UMAP should become a default companion to PCA in large genomic cohorts.

\matmethods{We used genotype data from 12,454 individuals from the Health and Retirement Study (HRS), genotyped on the Illumina Human Omni 2.5M platform\cite{juster1995overview}. Principal components were computed in PLINK v1.90b5.2 64-bit\cite{purcell2007plink} using variants with a minor allele frequency greater than 0.05, Hardy-Weinberg p-value of more than $1\times 10^{-6}$, and genotype missing rate of less than $0.1$, and sample with genotype missing rate of less than $0.1$. We used the principal components of genotype data from 488,377 individuals in the UK BioBank (UKBB) as computed by the cohort \cite{sudlow2015uk}. We used genotype data from 3,450 individuals from the 1KGP project using Affy 6.0 genotyping\cite{10002015global}.

The HRS contains genotype data of 12,454 American individuals across all 50 states who have provided racial identity (10,434 White, 1,652 Black, 368 Other) as well as whether they identify as Hispanic (1,203 total) and, if so, whether they identify as Mexican-American (705 total)\cite{juster1995overview}. We crossed these three variables to form a composite self-reported ethnicity resulting in 10 categories (e.g. White Hispanic Mexican-American), and considered birth regions based on the 10 census regions and divisions used by the US Census Bureau. Admixture proportions for each individual were estimated in \cite{baharian2016great} by assuming ancestral African, Asian, and European populations using RFMIX \cite{Maples:2013fia}. We have scaled each of the three proportions to values between 0 and 255 (with 100\% corresponding to 255), to colour individual points by their estimated admixture represented by RGB where red, green, and blue respectively correspond to African, European, and Asian/Native American ancestry. To project 1KGP data on HRS embeddings, as in figures~\ref{fig:supp_umap_hrs_hisp_1kgp_hisp} and \ref{fig:supp_hrs_1kgp_projected}, we created the PC axes and UMAP embedding for the HRS data and then projected the 1KGP data onto them.

The UKBB provides genotype data on 488,377 individuals along with self-identified ethnic background in a hierarchical tree-structured dictionary. Participants provided ethnic background on two occasions. We used the initial ethnicity after finding minimal differences between the two. The dataset is majority White (88.3\% British, 2.6\% Irish, 3.4\% other), with large populations identifying as Black (1.6\% either African, Caribbean, or other), Asian (1.9\% either Indian, Pakistani, Bangladeshi, or other), Chinese (0.3\%), an other ethnic group (0.8\%), mixed ethnicity (0.6\%), or an unavailable response (0.5\%). 

Scripts for all tests and plotting functions can be found on \url{https://github.com/diazale/gt-dimred}. A demo version using freely available 1KGP data is available at \url{https://github.com/diazale/1KGP_dimred}. PCA and standard t-SNE were done with Scikit-learn\cite{scikit-learn}. UMAP was performed using a Python implementation\cite{2018arXivUMAP}. Statistical testing was done in SciPy\cite{scipy} and StatsModels\cite{seabold2010statsmodels}.  

Both UMAP and t-SNE feature a number of adjustable parameters. Among the parameters that we varied, the number of PCs used in pre-processing of the data has the largest effect for both methods (see figures~\ref{fig:supp_megamontage_pc2_9} and \ref{fig:supp_megamontage_pc10_50}). 

We tested different choices for perplexity in t-SNE. The default value of 30 provided comparable performance to other parameter choices.  Similarly, we tested different parameter choices for UMAP, with the clearest results generated by specifying 15 nearest neighbours (the default value) and a "minimum distance" between points in low dimensions of 0.5. UMAP developers described "sensible" values for nearest neighbours as between 5 and 50 and minimum distance between 0.5 and 0.001. Tuning these parameters will not change qualitative results much but may make patterns easier to identify. Increasing the number of neighbours will increase the computational load, and a smaller minimum distance can break the connectivity between clusters, though the same individuals will continue to group together.

UMAP and t-SNE projections were carried out on an iMac with a 3.5GhZ Intel Core i7 processor, 32 GB 1600 MHz DDR3 of RAM, and an NVIDIA GeForce GTX 775M 2048 MB graphics card.

Colours for maps in figures~\ref{fig:umap_world_map}, \ref{fig:umap_ukbb_geo_map}, \ref{fig:supp_umap_ukbb_asia}, \ref{fig:supp_umap_ukbb_car}, and \ref{fig:supp_umap_ukbb_eur} were determined by projecting data to 3D and using each 3D coordinate as an RGB coordinate. For the world map, countries were determined using the country of birth variable, with a country's colour being determined by the mean $x$, $y$, and $z$ values of all individuals born in that country. Because many self-identified White British individuals were born abroad, including them everywhere would skew the colour scheme; they were included only if they were born in the UK, Europe, Australia, Canada, or the United States. This approach to colouring is sensitive to sample sizes as UMAP will give more space to larger populations. 

To reduce the potential risks for re-identification from results in this publication, data has been randomly permuted so that the population characteristics are preserved but individual-level data is not presented directly in the figures. We rounded each attribute to an attribute-specific number of bins, and then permuted the data in the following way: 
For each point (i.e. each individual) in UMAP visualizations, and each attribute, we identified the 9 nearest neighbouring points, and copied the attribute from a randomly selected neighbor (thus allowing for the possibility of one value being printed twice).  Because this process is done independently for each visualization, a given point shown on the figure will copy values from different randomly selected individuals. Projections coloured by participants' spatial coordinates have random noise added (normally distributed about 0 with a standard deviation of 50km) before binning to the nearest 50km. Projections coloured by participants' distance from London have random noise added (normally distributed about 0 with a standard deviation of 5km) before binning to the nearest km. For each point in figure~\ref{fig:umap_ukbb_geo_map} we identified the nearest 50 neighbouring individuals and copied the colour value from a randomly selected neighbour.}

\showmatmethods{} % Display the Materials and Methods section

%\acknow{}

%\showacknow{} % Display the acknowledgments section

\subsection*{Declarations}
\subsubsection*{Ethics approval and consent to participate}
HRS data was under IRB Study No. A11-E91-13B - The apportionment of genetic diversity within the United States. UKBB data was accessed under accession number 6728.

\subsubsection*{Consent for publication} Not applicable.

\subsubsection*{Availability of data and material} All data is publicly available to researchers.

\subsubsection*{Competing interests}
We have no competing interests to declare.

\subsubsection*{Funding}
This research was undertaken, in part, thanks to funding from the Canada Research Chairs program and CIHR grant MOP-136855.

\subsubsection*{Authors' contributions}
A.D.P. and S.G. designed the research. A.D.P. carried out dimension reduction and analysis. S.G. provided analysis. A.D.P and S.G. wrote the paper. L.A.T. provided the UK map visualization and analysis. C.B.E. prepared several UKBB datasets and principal component vectors.

\subsubsection*{Acknowledgements}
We thank all participants in the HRS, UKBB, and 1KGP for providing their genetic data as well as the teams who generated and assembled the datasets. We also thank Jose Sergio Hleap, Mark Lathrop, Dominic Nelson, Markus Munter, Stephen Sawcer, and Audrey Grant for useful discussions about science, programming, and data access; David Poznik, Liz Babalola, and Adam Auton from 23andMe for discussing findings in the 1KGP; and Selin Jessa for introducing us to UMAP. This research has been conducted using the UK Biobank Resource under Application Number 6728.

% Bibliography
\bibliography{pca_umap_genotype}

\newpage

\beginsupplement
\subsection*{Supporting Information (SI)}
\subsubsection*{SI Figures}
\begin{figure*}
    \centering
    \begin{subfigure}{\textwidth}
    \includegraphics[width=\textwidth]{images/megamontage_PC2_9.pdf}
    \end{subfigure}
    \caption{UMAP (left two columns) and t-SNE (right two columns) applied to the top principal components of the 1KGP labelled by the number of components used. Adding more components results in progressively finer population clusters using both methods.}
    \label{fig:supp_megamontage_pc2_9}  
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{0.95\textwidth}
    \includegraphics[width=0.95\textwidth]{images/megamontage_PC10_50.pdf}
    \end{subfigure}
    \caption{UMAP (left two columns) and t-SNE (right two columns) applied to the top principal components of the 1KGP labelled by the number of components used. Results are similar until approximately 11 components, where t-SNE breaks apart clusters of South Asian (in green) and Central and South American populations (in pink) while UMAP preserves them. At approximately 30 components populations begin to drift together with UMAP and disperse with t-SNE.}
    \label{fig:supp_megamontage_pc10_50}
\end{figure*}


\begin{figure*}
    \centering
    \begin{subfigure}{0.95\textwidth}
    \includegraphics[width=0.95\textwidth]{images/montage_1KGP_umap_convergence_resize.jpeg}
    \end{subfigure}
    \caption{UMAP applied to the first few hundred principal components of the 1KGP data with the amount of variance explained in parentheses. As more components are added, the figure begins to resemble that of UMAP carried out on the full genotype dataset.}
    \label{fig:supp_montage_1kgp_converge}
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{0.95\textwidth}
    \includegraphics[width=0.95\textwidth]{images/1KGP_UMAP_PCS100_PCE3450_NC2_NN15_MD05_201944185843.jpeg}
    \end{subfigure}
    \caption{UMAP applied the last 3350 principal components of the 1KGP, which explain 78.7\% of the variation.}
    \label{fig:supp_1kgp_3350}
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{\textwidth}
    \includegraphics[width=0.52\textwidth]{images/1KGP_UMAP_PC15_families.png}
    \end{subfigure}
    \caption{UMAP applied to the first 15 principal components of the 1KGP, with the number of neighbours set to 5 (top) and 15 (bottom). Six members of one Southern Han Chinese family are highlighted: HG00656 (grandfather), HG00657 (grandmother), HG00658 (uncle, mother's brother), HG00701 (mother), HG00702 (father), HG00703 (child). When using UMAP with five neighbours, the father (in blue) is projected to the cluster of the Southern Han Chinese population while the rest of the family members (in red) form their own disjoint cluster. Using 15 neighbours, the family still clusters together, but as part of the Southern Han Chinese population.}
    \label{fig:supp_1kgp_families}
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{\textwidth}
    \includegraphics[width=0.8\textwidth]{images/HRS_1000G_NP1_UMAP_PC7_NC2_NN15_MD05_pca_hrshisp_added1kgp_2018115153245_admix.pdf}
    \end{subfigure}
    \caption{UMAP of the first 7 principal components of the Hispanic population of the HRS, coloured by estimated admixture proportions. Admixture proportions for each individual were estimated in \cite{baharian2016great} by assuming ancestral African, Asian, and European populations using RFMIX \cite{Maples:2013fia}. We have scaled each of the three proportions to values between 0 and 255 (with 100\% corresponding to 255), to colour individual points by their estimated admixture represented by RGB where red, green, and blue respectively correspond to African, European, and Asian/Native American ancestry. An alternate colouring is provided in figure~\ref{fig:supp_umap_hrs_hisp_admix_alt}.}
    \label{fig:supp_umap_hrs_hisp_admix}
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{\textwidth}
    \includegraphics[width=0.8\textwidth]{images/HRS_1000G_NP1_UMAP_PC7_NC2_NN15_MD05_pca_hrshisp_added1kgp_2018115153245_admix132_hisp.jpeg}
    \end{subfigure}
    \caption{An alternate colouring of figure~\ref{fig:supp_umap_hrs_hisp_admix}. Here red, green, and blue correspond to African, Asian/Native American, and European ancestry, respectively.}
    \label{fig:supp_umap_hrs_hisp_admix_alt}
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{\textwidth}
    \includegraphics[width=0.8\textwidth]{images/HRS_1000G_NP1_UMAP_PC7_NC2_NN15_MD05_pca_hrshisp_added1kgp_2018115153245_1kgp_hisp_birthonly.jpeg}
    \end{subfigure}
    \caption{UMAP of the first 7 principal components of the Hispanic population of the HRS, coloured region of birth.}
    \label{fig:supp_umap_hrs_hisp_birth}
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{\textwidth}
    \includegraphics[width=0.8\textwidth]{images/Asthma_FullAsian_PC_QC_new_UMAP_PC5_NC2_NN15_MD05_201931173256.jpeg}
    \end{subfigure}
    \caption{UMAP of the first 8 principal components of the Asian population in the UKBB coloured by self-identified ethnicity. This is an alternate colouring of figure~\ref{fig:umap_ukbb_indian}}
    \label{fig:supp_umap_ukbb_asian_eth}
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{0.95\textwidth}
    \includegraphics[width=0.75\textwidth]{images/HRS_1000G_NP1_UMAP_PC10_NC2_NN15_MD05_pca_1kgp_onto_hrs_umap_1kgp_onto_hrs_2018112221116_race_hisp_mex_labels.pdf}
    \end{subfigure}
    \caption{UMAP applied to the first 10 principal components of HRS data. Points coloured by self-identified race, Hispanic status, and Mexican-American status. The cluster on the left is mostly people who identify as neither Black nor White and were born outside the contiguous United States or in the Pacific census region. Clustering with the 1KGP data places them with Asian-identified populations. BNH, Black (not Hispanic); BHO, Black (Hispanic, Other); WNH, White (not Hispanic); WHM, White (Hispanic, Mexican-American); WHO, White Hispanic (Other); ONH, Other (not Hispanic); OHM, Other (Hispanic, Mexican-American); OHO, Other (Hispanic, Other).}
    \label{fig:supp_umap_hrs_eth}
\end{figure*}

\begin{figure}
\centering
   \includegraphics[width=0.6\linewidth]{images/HRS_1000G_NP1_UMAP_PC10_NC2_NN15_MD05_pca_1kgp_onto_hrs_umap_1kgp_onto_hrs_2018112221116_admix.pdf}
   \caption{UMAP on the first 10 principal components of HRS data. colouring individuals by estimated admixture from three ancestral populations reveals considerable diversity in the Hispanic population. This projection coloured by self-identified race and Hispanic status is presented in figure~\ref{fig:supp_umap_hrs_eth}. Admixture proportions for each individual were estimated in \cite{baharian2016great} by assuming ancestral African, Asian, and European populations using RFMIX \cite{Maples:2013fia}. We have scaled each of the three proportions to values between 0 and 255 (with 100\% corresponding to 255), to colour individual points by their estimated admixture represented by RGB where red, green, and blue respectively correspond to African, European, and Asian/Native American ancestry. An alternate colouring is provided in figure~\ref{fig:umap_hrs_admix_alt}}
    \label{fig:umap_hrs_admix}
\end{figure}

\begin{figure}
\centering
   \includegraphics[width=0.6\linewidth]{images/HRS_1000G_NP1_UMAP_PC10_NC2_NN15_MD05_pca_1kgp_onto_hrs_umap_1kgp_onto_hrs_2018112221116_admix132.jpeg}
   \caption{An alternate colouring of \ref{fig:umap_hrs_admix}. Here red, green, and blue correspond to African, Asian/Native American, and European ancestry, respectively.}
    \label{fig:umap_hrs_admix_alt}
\end{figure}

\begin{figure*}
    \centering
    \begin{subfigure}{\textwidth}
    \includegraphics[width=0.75\textwidth]{images/HRS_1000G_NP1_UMAP_PC10_NC2_NN15_MD05_pca_1kgp_onto_hrs_umap_1kgp_onto_hrs_2018112221116_born.pdf}
    \end{subfigure}
    \caption{UMAP on the top 10 principal components of the HRS dataset, coloured by Census Bureau birth region. Each colour represents one of the 10 birth regions. There is no obvious pattern in the clusters of majority "White Not Hispanic" individuals.}
    \label{fig:supp_hrs_born}
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{\textwidth}
    \includegraphics[width=\textwidth]{images/HRS_1000G_UMAP_PC10_NC2_NN15_MD05_2018627203416_label.pdf}
    \end{subfigure}
    \caption{UMAP projection of the top 10 principal components of the combined HRS and 1KGP datasets. One cluster (in the box) does not group with any of the 1KGP populations. A cluster of Finnish (FIN) individuals consistently appears in the "White Not Hispanic" (WNH) group. Groups of Central and South American populations from the 1KGP (CLM, Colombian; MXL, Mexican; PEL, Peruvian; PUR, Puerto Rican) form nearby or within the HRS Hispanic cluster (HIS). Iberian individuals (IBS) cluster near the Hispanic population. Toscani individuals (TSI) form some small clusters and sometimes appear near the Iberian and Hispanic populations. Individuals with British/Scottish (GBR) or Northern/Western European ancestry (CEU) are scattered throughout the WNH clusters. Individuals with African ancestry from the 1KGP group with Black Americans from the HRS (AFR). Similar population groupings occur with South Asian (SAS) and East Asian (EAS) individuals.}
    \label{fig:supp_hrs_1000g}
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{\textwidth}
    \includegraphics[width=\textwidth]{images/HRS_1000G_NP1_UMAP_PC10_NC2_NN15_MD05_pca_1kgp_onto_hrs_umap_1kgp_onto_hrs_2018112221116_custom_label.pdf}
    \end{subfigure}
    \caption{UMAP on the top 10 principal components of the HRS data, with 1KGP data projected onto the embedding. Individuals from the HRS are grey. British (GBR) and other European (CEU) individuals are scattered throughout the "White Not Hispanic" clusters. Finns (FIN) form clear groupings. Spanish (IBS) and Italian (TSI) individuals cluster near the Hispanic grouping. There are sub-groups in the Hispanic cluster formed of Puerto Ricans (PUR), Colombians (CLM), Mexicans (MXL), and Peruvians (PEL). Populations with African ancestry (AFR) appear with Black individuals. East Asian (EAS) populations comprising Chinese, Kinh, and Japanese individuals cluster together with what appears in figure~\ref{fig:umap_hrs_admix} as a population of mostly Asian ancestry. South Asian (SAS) populations with Indian, Pakistani, and Sri Lankan ancestry cluster in a separate area. One "White Not Hispanic" cluster at the bottom does not cluster with any 1KGP populations.}
    \label{fig:supp_hrs_1kgp_projected}
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{\textwidth}
    \includegraphics[width=\textwidth]{images/HRS_PCGRID_8.pdf}
    \end{subfigure}
    \caption{Pairwise plots of the first 8 principal components of the Hispanic subset of the HRS. Those born in the Mountain region are coloured green.}
    \label{fig:supp_hrs_hisp_grid}
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{0.8\textwidth}
    \includegraphics[width=\textwidth]{images/UKBB_UMAP_PC10_NN15_MD05_eth_cob_resized.jpeg}
    \end{subfigure}
    \caption{Using country of birth data, some of the larger unidentified groups from figure~\ref{fig:supp_ukbb_zoom} were identified as being born mostly in Japan, the Philippines, North Africa, the Middle East, and Central and South America. The large cluster of "Any other Asian Background" were mostly born in Sri Lanka.}
    \label{fig:supp_ukbb_cob}
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{0.9\textwidth}
    \includegraphics[width=0.9\textwidth]{images/UKBB_UMAP_PC10_NN15_MD05_2018328174511_london_permuted_10nn_sd10_base1_maxdist200_2019625172227_resize.jpeg}
    \end{subfigure}
    \caption{UMAP on UKBB data, coloured by distance from London, with red representing those living closer to London and blue representing those living farther from London. A 200km radius extends roughly to Cardiff, and a 100km radius extends roughly to cities such as Leicester and Bath, and contains cities such as Oxford, Cambridge, and Peterborough. Data has been randomized as explained in the materials and methods section.}
    \label{fig:supp_london_distance}
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{0.95\textwidth}
    \includegraphics[width=0.95\textwidth]{images/default_clean_size10_alpha60_flip.jpeg}
    \end{subfigure}
    \caption{UMAP on UKBB data, coloured by self-identified ethnic background. Images are labelled by the number of components included.}
    \label{fig:supp_montage_ukbb_eth}
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{0.95\textwidth}
    \includegraphics[width=0.95\textwidth]{images/default_clean_size10_alpha60_ns.jpeg}
    \end{subfigure}
    \caption{UMAP on UKBB data, coloured by northing values, with more blue representing more northern coordinates and more red representing more southern coordinates. Images are labelled by the number of components included. Data has been randomized as explained in the materials and methods section.}
    \label{fig:supp_montage_ukbb_ns}
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{\textwidth}
    \includegraphics[width=\textwidth]{images/default_clean_size10_alpha60_ew.jpeg}
    \end{subfigure}
    \caption{UMAP on UKBB data, coloured by easting values, with more yellow representing more eastern coordinates and more pink representing more western coordinates. Images are labelled by the number of components included. Data has been randomized as explained in the materials and methods section.}
    \label{fig:supp_montage_ukbb_ew}
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{\textwidth}
    \includegraphics[width=\textwidth]{images/meanAsiaMap.jpg}
    \end{subfigure}
    \caption{Figure~\ref{fig:umap_world_map}, zoomed in on Asia. Geographic distribution of UMAP coordinates. Using the country of birth of individuals in the UKBB, we colour countries by the closeness in 3D UMAP space of those born there. Broad patterns of similarity appear in East Asia, South Asia, North African and the Middle East, West Africa, and South America. Differences between neighbouring countries can reflect both ancient population structure and recent differences in migration history. Evidence of migrations related to colonialism are visible with, e.g., European ancestry in South Africa and South Asian ancestry in Kenya and Tanzania. Because of the large number of White British individuals born abroad, to avoid skewing the colour scale they were not included unless they were born in the UK, Europe, Australia, Canada, or the United States, where UKBB participants already tended to have European ancestry.}
    \label{fig:supp_umap_ukbb_asia}
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{\textwidth}
    \includegraphics[width=\textwidth]{images/meanCarribeanMap.jpg}
    \end{subfigure}
    \caption{Figure~\ref{fig:umap_world_map}, zoomed in on the Caribbean. Geographic distribution of UMAP coordinates. Using the country of birth of individuals in the UKBB, we colour countries by the closeness in 3D UMAP space of those born there. Broad patterns of similarity appear in East Asia, South Asia, North African and the Middle East, West Africa, and South America. Differences between neighbouring countries can reflect both ancient population structure and recent differences in migration history. Evidence of migrations related to colonialism are visible with, e.g., European ancestry in South Africa and South Asian ancestry in Kenya and Tanzania. Because of the large number of White British individuals born abroad, to avoid skewing the colour scale they were not included unless they were born in the UK, Europe, Australia, Canada, or the United States, where UKBB participants already tended to have European ancestry.}
    \label{fig:supp_umap_ukbb_car}
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{\textwidth}
    \includegraphics[width=\textwidth]{images/meanEuroMap.jpg}
    \end{subfigure}
    \caption{Figure~\ref{fig:umap_world_map}, zoomed in on Europe. Geographic distribution of UMAP coordinates. Using the country of birth of individuals in the UKBB, we colour countries by the closeness in 3D UMAP space of those born there. Broad patterns of similarity appear in East Asia, South Asia, North African and the Middle East, West Africa, and South America. Differences between neighbouring countries can reflect both ancient population structure and recent differences in migration history. Evidence of migrations related to colonialism are visible with, e.g., European ancestry in South Africa and South Asian ancestry in Kenya and Tanzania. Because of the large number of White British individuals born abroad, to avoid skewing the colour scale they were not included unless they were born in the UK, Europe, Australia, Canada, or the United States, where UKBB participants already tended to have European ancestry. }
    \label{fig:supp_umap_ukbb_eur}
\end{figure*}

\begin{figure}[!htb]
    \centering
    \includegraphics[width=0.95\columnwidth]{images/UKBB_TSNE_10PCs_DefaultPerplexity_eth.pdf}
    \caption{t-SNE applied to the top 10 principal components of the UKBB, coloured by ethnic background. The unbalanced populations resulted in many individuals and populations being orphaned along the periphery of the main cluster.}
    \label{fig:supp_ukbb_tsne}
\end{figure}

%%%%% UKBB phenotype plots
\begin{figure}
    \centering
    %\begin{subfigure}{0.8\textwidth}
    \includegraphics[width=0.4\columnwidth]{images/UKBB_UMAP_PC10_NN15_MD05_2018328174511_201871417039_basophill_count_pct5_f.pdf}
    %\end{subfigure}
    \caption{UMAP on the top 10 principal components of the UKBB coloured by basophil count (female). Data has been randomized as explained in the materials and methods section.}
    \label{fig:supp_ukbb_basophill_f}
\end{figure}

\begin{figure}
    \centering
    %\begin{subfigure}{0.8\textwidth}
    \includegraphics[width=0.4\columnwidth]{images/UKBB_UMAP_PC10_NN15_MD05_2018328174511_201871417039_basophill_count_pct5_m.pdf}
    %\end{subfigure}
    \caption{UMAP on the top 10 principal components of the UKBB coloured by basophil count (male). Data has been randomized as explained in the materials and methods section.}
    \label{fig:supp_ukbb_basophill_m}
\end{figure}

\begin{figure}
    \centering
    %\begin{subfigure}{0.8\textwidth}
    \includegraphics[width=0.4\columnwidth]{images/UKBB_UMAP_PC10_NN15_MD05_2018328174511_201871417720_eosinophill_count_pct5_f.pdf}
    %\end{subfigure}
    \caption{UMAP on the top 10 principal components of the UKBB coloured by eosinophil count (female). Data has been randomized as explained in the materials and methods section.}
    \label{fig:supp_ukbb_eosinophill_f}
\end{figure}

\begin{figure}
    \centering
    %\begin{subfigure}{0.8\textwidth}
    \includegraphics[width=0.4\columnwidth]{images/UKBB_UMAP_PC10_NN15_MD05_2018328174511_201871417720_eosinophill_count_pct5_m.pdf}
    %\end{subfigure}
    \caption{UMAP on the top 10 principal components of the UKBB coloured by eosinophil count (male). Data has been randomized as explained in the materials and methods section.}
    \label{fig:supp_ukbb_eosinophill_m}
\end{figure}

\begin{figure}
    \centering
    %\begin{subfigure}{0.8\textwidth}
    \includegraphics[width=0.4\columnwidth]{images/UKBB_UMAP_PC10_NN15_MD05_2018328174511_201871416305_3063_0_0_pct1_f.pdf}
    %\end{subfigure}
    \caption{UMAP on the top 10 principal components of the UKBB coloured by FEV1 (female). Data has been randomized as explained in the materials and methods section.}
    \label{fig:supp_ukbb_fev_f}
\end{figure}

\begin{figure}
    \centering
    %\begin{subfigure}{0.8\textwidth}
    \includegraphics[width=0.4\columnwidth]{images/UKBB_UMAP_PC10_NN15_MD05_2018328174511_201871416305_3063_0_0_pct1_m.pdf}
    %\end{subfigure}
    \caption{UMAP on the top 10 principal components of the UKBB coloured by FEV1 (male). Data has been randomized as explained in the materials and methods section.}
    \label{fig:supp_ukbb_fev_m}
\end{figure}

\begin{figure}
    \centering
    %\begin{subfigure}{0.8\textwidth}
    \includegraphics[width=0.4\columnwidth]{images/UKBB_UMAP_PC10_NN15_MD05_2018328174511_2018714161841_Height_res_pct1_f.pdf}
    %\end{subfigure}
    \caption{UMAP on the top 10 principal components of the UKBB coloured by height (female). Data has been randomized as explained in the materials and methods section.}
    \label{fig:supp_ukbb_height_f}
\end{figure}

\begin{figure}
    \centering
    %\begin{subfigure}{0.8\textwidth}
    \includegraphics[width=0.4\columnwidth]{images/UKBB_UMAP_PC10_NN15_MD05_2018328174511_2018714161841_Height_res_pct1_m.pdf}
    %\end{subfigure}
    \caption{UMAP on the top 10 principal components of the UKBB coloured by height (male). Data has been randomized as explained in the materials and methods section.}
    \label{fig:supp_ukbb_height_m}
\end{figure}

\begin{figure}
    \centering
    %\begin{subfigure}{0.8\textwidth}
    \includegraphics[width=0.4\columnwidth]{images/UKBB_UMAP_PC10_NN15_MD05_2018328174511_201871416519_leukocyte_count_pct5_f.pdf}
    %\end{subfigure}
    \caption{UMAP on the top 10 principal components of the UKBB coloured by leukocyte count (female). Data has been randomized as explained in the materials and methods section.}
    \label{fig:supp_ukbb_leukocyte_f}
\end{figure}

\begin{figure}
    \centering
    %\begin{subfigure}{0.8\textwidth}
    \includegraphics[width=0.4\columnwidth]{images/UKBB_UMAP_PC10_NN15_MD05_2018328174511_201871416519_leukocyte_count_pct5_m.pdf}
    %\end{subfigure}
    \caption{UMAP on the top 10 principal components of the UKBB coloured by leukocyte count (male). Data has been randomized as explained in the materials and methods section.}
    \label{fig:supp_ukbb_leukocyte_m}
\end{figure}

\begin{figure}
    \centering
    %\begin{subfigure}{0.8\textwidth}
    \includegraphics[width=0.4\columnwidth]{images/UKBB_UMAP_PC10_NN15_MD05_2018328174511_2018714165614_neutrophill_count_pct5_f.pdf}
    %\end{subfigure}
    \caption{UMAP on the top 10 principal components of the UKBB coloured by neutrophil count (female). Data has been randomized as explained in the materials and methods section.}
    \label{fig:supp_ukbb_neutrophill_f}
\end{figure}

\begin{figure}
    \centering
    %\begin{subfigure}{0.8\textwidth}
    \includegraphics[width=0.4\columnwidth]{images/UKBB_UMAP_PC10_NN15_MD05_2018328174511_2018714165614_neutrophill_count_pct5_m.pdf}
    %\end{subfigure}
    \caption{UMAP on the top 10 principal components of the UKBB coloured by neutrophil count (male). Data has been randomized as explained in the materials and methods section.}
    \label{fig:supp_ukbb_neutrophill_m}
\end{figure}

%%%%% UKBB phenotypes end

\begin{figure*}
    \centering
    %\begin{subfigure}{0.8\textwidth}
    \includegraphics[width=0.4\columnwidth]{images/ukbb_pcs_2019410184041_Height_res_pct1_f.jpeg}
    %\end{subfigure}
    \caption{Principal components 1 and 2 from the UKBB, coloured by age-adjusted residual height (female). Data has been randomized as explained in the materials and methods section.}
    \label{fig:supp_ukbb_pca_height_res_f}
\end{figure*}

\begin{figure*}
    \centering
    %\begin{subfigure}{0.8\textwidth}
    \includegraphics[width=0.4\columnwidth]{images/ukbb_pcs_201941019249_3063_0_0_pct1_f.jpeg}
    %\end{subfigure}
    \caption{Principal components 1 and 2 from the UKBB, coloured by FEV1 (female). Data has been randomized as explained in the materials and methods section.}
    \label{fig:supp_ukbb_pca_fev_f}
\end{figure*}

%\begin{figure*}
%    \centering
%    \includegraphics[width=0.4\columnwidth]{images/ukbb_pcs_201941019249_3063_0_0_pct1_f.jpeg}
%    \caption{Test. Data has been randomized as explained in the materials and methods section.}
%    \label{fig:supp_ukbb_tsne_fev_f}
%\end{figure*}

\clearpage
\begin{figure*}
    \centering
    %\begin{subfigure}{0.8\textwidth}
    \includegraphics[width=0.4\columnwidth]{images/UKBB_TSNE_10PCs_DefaultPerplexity_2019410184913_Height_res_pct1_f.jpeg}
    %\end{subfigure}
    \caption{t-SNE on the first 10 principal components from the UKBB, coloured by age-adjusted residual height (female). Data has been randomized as explained in the materials and methods section.}
    \label{fig:supp_ukbb_tsne_height_res_f}
\end{figure*}

\begin{figure*}
    \centering
    %\begin{subfigure}{0.8\textwidth}
    \includegraphics[width=0.4\columnwidth]{images/UKBB_TSNE_10PCs_DefaultPerplexity_2019410191255_3063_0_0_pct1_f.jpeg}
    %\end{subfigure}
    \caption{t-SNE on the first 10 principal components from the UKBB, coloured by FEV1 (female). Data has been randomized as explained in the materials and methods section.}
    \label{fig:supp_ukbb_tsne_fev_f}
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{0.8\textwidth}
    \includegraphics[width=\textwidth]{images/UKBB_UMAP_PC10_NN15_MD05_eth_combined_resized.pdf}
    \end{subfigure}
    \caption{Zoomed in areas of figure~\ref{fig:umap_ukbb_pc10}. Sections (i) and (ii) respectively focus on the African and Asian superpopulations, and section (iii) focuses on an area with individuals from many ethnic backgrounds. Noticeable clusters of unidentified ethnic backgrounds appear and are labelled "OEG" "(Other Ethnic Group)".}
    \label{fig:supp_ukbb_zoom}
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{\textwidth}
    \includegraphics[width=\textwidth]{images/montage_fev1_height_afr_permuted.pdf}
    \end{subfigure}
    \caption{Individuals of Black African, Black Caribbean, and mixed backgrounds (primarily White and Black Caribbean/African) coloured by self-identified ethnic background (left, from figure~\ref{fig:umap_ukbb_pc10}), FEV1 (middle), and age-adjusted height (right). An arrow points to an area where the FEV1 distribution appears to change, corresponding to where the clusters contain more people with self-identified mixed backgrounds.}
    \label{fig:supp_comparison_fev_afr}
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{\textwidth}
    \includegraphics[width=\textwidth]{images/montage_fev1_height_chi_eur_permuted.pdf}
    \end{subfigure}
    \caption{Zoomed in section of figure~\ref{fig:umap_ukbb_pc10} focused on individuals with Chinese (CHI), White British (GBR), any other white background, or any other ethnic group (OEG) coloured by ethnicity (left), FEV1 (middle), and age-adjusted height (right). The OEG cluster next to the Chinese cluster is coloured differently, suggesting this population may have different FEV1 characteristics. A cluster of OEG/other white individuals is more blue, suggesting they may have lower than average FEV1 values relative to the rest of the British or white population.}
    \label{fig:supp_comparison_fev_chi_eur}
\end{figure*}



\begin{figure*}
    \centering
    \begin{subfigure}{0.5\textwidth}
    \includegraphics[width=0.5\textwidth]{images/east_asia_population_colours.png}
    \end{subfigure}
    \caption{Individuals from the zoomed in section in figure~\ref{fig:supp_comparison_fev_chi_eur} used in statistical testing, coloured the same as in figure~\ref{fig:supp_fev_ridgeplots}. Brown, blue, and green represent those born in the Philippines, Malaysia, and Japan; pink represents those who self-identify as Chinese. The Chinese individuals were those who self-identified their ethnic background as Chinese, and the remaining populations were determined based on country of birth; the categorizations are mutually exclusive.}
    \label{fig:supp_fev_test_pops}
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{\textwidth}
    \includegraphics[width=\textwidth]{images/east_asia_ridgeplots_fev.png}
    \end{subfigure}
    \caption{Plots of the distributions of residual FEV1 by sex for East Asian populations, after adjusting for height, age, age$^2$, and sex through linear regression. Individuals were limited to those in the "Chinese/Other Ethnic Group" cluster from figure~\ref{fig:supp_comparison_fev_chi_eur}. The Chinese individuals were those who self-identified their ethnic background as Chinese, and the remaining populations were determined based on country of birth; the categorizations are mutually exclusive. Asterisks indicate significant difference from the Japanese population, using Welch's unpaired t-test with a Bonferroni correction for 3 groups. The dashed lines are the means of the distributions, and Japanese populations have consistently higher means.}
    \label{fig:supp_fev_ridgeplots}
\end{figure*}

\begin{figure}[!htb]
    \centering
    \includegraphics[width=0.95\columnwidth]{images/1KGP_tsne_umap.jpeg}
    \caption{Comparing the visualizations of UMAP, standard t-SNE, and t-SNE initialized with a UMAP projection, on the top 10 principal components of the 1KGP. t-SNE used 5000 iterations. Initializing t-SNE with UMAP breaks the continuous structure of the projection and instead forms many small clusters.}
    \label{fig:supp_tsne_umap_compare_1kgp}
\end{figure}

\begin{figure}[!htb]
    \centering
    \includegraphics[width=0.95\columnwidth]{images/HRS_tsne_umap.jpeg}
    \caption{Comparing the visualizations of UMAP, standard t-SNE, and t-SNE initialized with a UMAP projection, on the top 10 principal components of the HRS. t-SNE used 5000 iterations.}
    \label{fig:supp_tsne_umap_compare_hrs}
\end{figure}

\begin{figure}[!htb]
    \centering
    \includegraphics[width=0.95\columnwidth]{images/tsne_umap_graph_ukbb.jpeg}
    \caption{Comparing the error terms of standard t-SNE versus t-SNE initialized with a UMAP embedding and no early exaggeration. Done on the UKBB dataset with 20000 iterations. The UMAP-initialized graph has been shifted by 230 iterations to approximate the 230 epochs UMAP uses for large datasets ($n>10,000$).}
    \label{fig:supp_tsne_umap_compare_ukbb_graph}
\end{figure}

\begin{figure}[!htb]
    \centering
    \includegraphics[width=0.95\columnwidth]{images/ukbb_tsne_umap.jpeg}
    \caption{Comparing the visualizations of UMAP, standard t-SNE, and t-SNE initialized with a UMAP projection, on the top 10 principal components of the UKBB. t-SNE used 20000 iterations}
    \label{fig:supp_tsne_umap_compare_ukbb}
\end{figure}

\begin{figure}[!htb]
    \centering
    \includegraphics[width=0.95\columnwidth]{images/tsne_umap_graph_1kgp.jpeg}
    \caption{Comparing the error terms of standard t-SNE versus t-SNE initialized with a UMAP embedding and no early exaggeration. Done on the 1KGP dataset with 5000 iterations. The UMAP-initialized graph has been shifted by 600 iterations to approximate the 600 epochs UMAP uses for small datasets ($n\leq10,000$).}
    \label{fig:supp_tsne_umap_compare_1kgp_graph}
\end{figure}

\begin{figure}[!htb]
    \centering
    \includegraphics[width=0.95\columnwidth]{images/tsne_umap_graph_hrs.jpeg}
    \caption{Comparing the error terms of standard t-SNE versus t-SNE initialized with a UMAP embedding and no early exaggeration. Done on the HRS dataset with 5000 iterations. The UMAP-initialized graph has been shifted by 230 iterations to approximate the 230 epochs UMAP uses for large datasets ($n>10,000$).}
    \label{fig:supp_tsne_umap_compare_hrs_graph}
\end{figure}



%%%%% UKBB boxplots below

\begin{figure*}
    \centering
    \begin{subfigure}{\textwidth}
    \includegraphics[width=\textwidth]{images/female_basophill_boxplot_annotated.pdf}
    \end{subfigure}
    \caption{Basophil counts by sex and ethnic group, annotated with p-values. Asterisks indicate significant difference from the White British group with a Bonferroni correction for 12 groups.}
    \label{fig:supp_box_basophill_f}
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{\textwidth}
    \includegraphics[width=\textwidth]{images/male_basophill_boxplot_annotated.pdf}
    \end{subfigure}
    \caption{Basophil counts by sex and ethnic group, annotated with p-values. Asterisks indicate significant difference from the White British group with a Bonferroni correction for 12 groups.}
    \label{fig:supp_box_basophill_m}
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{\textwidth}
    \includegraphics[width=\textwidth]{images/female_eosinophill_boxplot_annotated.pdf}
    \end{subfigure}
    \caption{Eeosinophil counts by sex and ethnic group, annotated with p-values. Asterisks indicate significant difference from the White British group with a Bonferroni correction for 12 groups.}
    \label{fig:supp_box_eosinophill_f}
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{\textwidth}
    \includegraphics[width=\textwidth]{images/male_eosinophill_boxplot_annotated.pdf}
    \end{subfigure}
    \caption{Eosinophil counts by sex and ethnic group, annotated with p-values. Asterisks indicate significant difference from the White British group with a Bonferroni correction for 12 groups.}
    \label{fig:supp_box_eosinophill_m}
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{\textwidth}
    \includegraphics[width=\textwidth]{images/female_fev_boxplot_annotated.pdf}
    \end{subfigure}
    \caption{FEV1 by sex and ethnic group, annotated with p-values. Asterisks indicate significant difference from the White British group with a Bonferroni correction for 12 groups.}
    \label{fig:supp_box_fev_f}
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{\textwidth}
    \includegraphics[width=\textwidth]{images/male_fev_boxplot_annotated.pdf}
    \end{subfigure}
    \caption{FEV1 by sex and ethnic group, annotated with p-values. Asterisks indicate significant difference from the White British group with a Bonferroni correction for 12 groups.}
    \label{fig:supp_box_fev_m}
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{\textwidth}
    \includegraphics[width=\textwidth]{images/female_height_boxplot_annotated.pdf}
    \end{subfigure}
    \caption{Height by sex and ethnic group, annotated with p-values. Asterisks indicate significant difference from the White British group with a Bonferroni correction for 12 groups.}
    \label{fig:supp_box_height_f}
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{\textwidth}
    \includegraphics[width=\textwidth]{images/male_height_boxplot_annotated.pdf}
    \end{subfigure}
    \caption{Height by sex and ethnic group, annotated with p-values. Asterisks indicate significant difference from the White British group with a Bonferroni correction for 12 groups.}
    \label{fig:supp_box_height_m}
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{\textwidth}
    \includegraphics[width=\textwidth]{images/female_leukocyte_boxplot_annotated.pdf}
    \end{subfigure}
    \caption{Leukocyte counts by sex and ethnic group, annotated with p-values. Asterisks indicate significant difference from the White British group with a Bonferroni correction for 12 groups.}
    \label{fig:supp_box_leukocyte_f}
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{\textwidth}
    \includegraphics[width=\textwidth]{images/male_leukocyte_boxplot_annotated.pdf}
    \end{subfigure}
    \caption{Leukocyte counts by sex and ethnic group, annotated with p-values. Asterisks indicate significant difference from the White British group with a Bonferroni correction for 12 groups.}
    \label{fig:supp_box_leukocyte_m}
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{\textwidth}
    \includegraphics[width=\textwidth]{images/female_neutrophil_boxplot_annotated.pdf}
    \end{subfigure}
    \caption{Neutrophil counts by sex and ethnic group, annotated with p-values. Asterisks indicate significant difference from the White British group with a Bonferroni correction for 12 groups.}
    \label{fig:supp_box_neutrophill_f}
\end{figure*}

\begin{figure*}
    \centering
    \begin{subfigure}{\textwidth}
    \includegraphics[width=\textwidth]{images/male_neutrophil_boxplot_annotated.pdf}
    \end{subfigure}
    \caption{Neutrophil counts by sex and ethnic group, annotated with p-values. Asterisks indicate significant difference from the White British group with a Bonferroni correction for 12 groups.}
    \label{fig:supp_box_neutrophill_m}
\end{figure*}

%%%%% boxplots end
%\begin{figure*}
%    \centering
%    \begin{subfigure}{\textwidth}
%    \includegraphics[width=\textwidth]{images/meanWorldMap_mindist001.jpg}
%    \end{subfigure}
%    \caption{Alternate colouring of figure~\ref{fig:umap_world_map} using a UMAP projection on the top 40 components with minimum distance 0.001.}
%    \label{fig:supp_umap_ukbb_worldmap_alt}
%\end{figure*}


\begin{figure*}
    \centering
    \begin{subfigure}{\textwidth}
    \includegraphics[width=\textwidth]{images/admixture_plot_highlight_mountain_copy.pdf}
    \end{subfigure}
    \caption{Admixture plot of Hispanic individuals in the HRS. Individuals born in the Mountain census region fall between the white lines (indices 48 to 184)}
    \label{fig:supp_hrs_hisp_admix}
\end{figure*}

\clearpage
\begin{table}[htb!]
\centering
 \begin{tabular}{||c c||} 
 \hline
 Number of PCs & Variance explained  \\ [0.5ex] 
 \hline\hline
 2 & 13.6\% \\
\hline
3 & 14.7\% \\
\hline
4 & 15.5\% \\
\hline
5 & 15.7\% \\
\hline
6 & 15.8\% \\
\hline
7 & 15.9\% \\
\hline
8 & 16.0\% \\
\hline
9 & 16.1\% \\
\hline
10 & 16.2\% \\
\hline
11 & 16.2\% \\
\hline
12 & 16.3\% \\
\hline
13 & 16.4\% \\
\hline
14 & 16.5\% \\
\hline
15 & 16.5\% \\
\hline
30 & 17.5\% \\
\hline
50 & 18.6\% \\
\hline
100 & 21.3\% \\
\hline
200 & 26.4\% \\
\hline
300 & 31.1\% \\
\hline
400 & 35.5\% \\
\hline
500 & 39.7\% \\
\hline
600 & 43.7\% \\
\hline
700 & 47.4\% \\
\hline
800 & 50.8\% \\
\hline
900 & 54.0\% \\
\hline
1000 & 57.0\% \\
\hline
1100 & 60.0\% \\
\hline
1200 & 62.8\% \\
\hline
1300 & 65.4\% \\
\hline
1400 & 68.0\% \\
\hline
1500 & 70.5\% \\
\hline
1600 & 73.0\% \\
\hline
1700 & 75.3\% \\
\hline
1800 & 77.5\% \\
\hline
1900 & 79.7\% \\
\hline
2000 & 81.8\% \\
\hline
2100 & 83.8\% \\
\hline
2200 & 85.8\% \\
\hline
2300 & 87.6\% \\
\hline
2400 & 89.4\% \\
\hline
2500 & 91.1\% \\
\hline
2600 & 92.7\% \\
\hline
2700 & 94.2\% \\
\hline
2800 & 95.3\% \\
\hline
2900 & 96.3\% \\
\hline
3000 & 97.1\% \\
\hline
3100 & 97.8\% \\
\hline
3200 & 98.5\% \\
\hline
3300 & 99.2\% \\
\hline
3400 & 99.7\% \\[1ex] 
 \hline
 \end{tabular}
 \caption{Variance explained in the 1KGP data by the number of principal components used.}
 \label{table:1KGP_var_exp}
\end{table}


\end{document}

